{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "import numpy as np\n",
    "import tensorflow as tf\n",
    "from tensorflow import keras\n",
    "### hack tf-keras to appear as top level keras\n",
    "import sys\n",
    "sys.modules['keras'] = keras\n",
    "### end of hack\n",
    "from keras import layers\n",
    "from keras.models import Model\n",
    "import keras.backend as K\n",
    "import json\n",
    "import gensim\n",
    "import os"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "import re\n",
    "from sklearn.model_selection import train_test_split\n",
    "from sklearn.metrics.pairwise import cosine_similarity\n",
    "import objectpath"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "home = os.getcwd()\n",
    "dataDir = home + '/bioasq/data'\n",
    "modelsDir = home + '/bioasq/models/dan.h5'\n",
    "embeddingFile = home + '/bioasq/embeddings/pubmed2018_w2v_200D.bin'\n",
    "trainFilePath = dataDir + '/BioASQ-trainingDataset6b.json'\n",
    "trainJSON = {}\n",
    "no_vectors = {}\n",
    "yes_vectors = {}\n",
    "missing_vectors = {}\n",
    "embedding_dimension = 200\n",
    "labels = {\n",
    "    \"summary\": [1, 0, 0, 0],\n",
    "    \"list\":    [0, 1, 0, 0],\n",
    "    \"yesno\":   [0, 0, 1, 0],\n",
    "    \"factoid\": [0, 0, 0, 0]\n",
    "}\n",
    "label_dim = 4"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Load data from files\n",
    "pubmedW2V = gensim.models.KeyedVectors.load_word2vec_format(embeddingFile, binary=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [],
   "source": [
    "with open(trainFilePath) as f:\n",
    "    trainJSON = json.load(f)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [],
   "source": [
    "def seperate_punctuation(str):\n",
    "    pStr = str.lower()\n",
    "    pStr = \" \".join(re.findall(r\"[\\w']+|[.,!?;'s]\", pStr))\n",
    "    remove_chars = [\"'s\", \"'t\", \"s'\", \"'\", \",\", \".\",\"?\", \"!\"]\n",
    "    for ch in remove_chars:\n",
    "        pStr = pStr.replace(ch, \"\")\n",
    "    pStr = pStr.replace(\"  \", \" \")\n",
    "    return pStr"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'ret gdnf ednrb edn3  and sox10 lead to long segment l hscr '"
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "seperate_punctuation(\"RET, GDNF, EDNRB, EDN3!, ands' SOX10 lead's to't long-segment (L-HSCR)?\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [],
   "source": [
    "\n",
    "def getQuestionembeddingsFromText(qText):\n",
    "    global no_vectors\n",
    "    global yes_vectors\n",
    "    global missing_vectors\n",
    "    qVect = seperate_punctuation(qText)\n",
    "    qWords = qVect.split(\" \")\n",
    "    qEmbeddings = np.zeros((len(qWords), embedding_dimension))\n",
    "    for i, qWord in enumerate(qWords):\n",
    "        qEmbedding = np.zeros((1, embedding_dimension))\n",
    "        try:\n",
    "            qEmbeddings[i:] = pubmedW2V.get_vector(qWord)  \n",
    "            if qWord not in yes_vectors:\n",
    "                yes_vectors[qWord] = 0;\n",
    "            yes_vectors[qWord] += 1\n",
    "        except:\n",
    "            if qWord not in no_vectors:\n",
    "                no_vectors[qWord] = 0;\n",
    "            if qWord not in missing_vectors:\n",
    "                missing_vectors[qWord] = np.random.randn(1, embedding_dimension)\n",
    "            no_vectors[qWord] += 1\n",
    "            qEmbeddings[i:] = missing_vectors[qWord]        \n",
    "    qEmbeddings = np.mean(qEmbeddings, axis=0, keepdims=True)\n",
    "    return qEmbeddings"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "   levoxyl   mirduplexsvm   woolsorter   mepopausal   itagliptin   turbeculosis   pregrancy   6srna   axagliptin   dediodinases   alterred   phopspholamban   mer41   path2ppi   mesaure   sfpr3   hydroxyisobutyrylation   ranasmurfin   metazaon   immonostaining   chédiak   secreatase   thyroitoxicosis   hisrchsprung   regioner   bouble   menzerath   kub5   archaelogy   acetylgalactosaminidase   glycolylneuraminic   arcalyst   subtahalamic   μac1   sqtlseeker   trigered   phoshpatases\n",
      "\n",
      "\n",
      "No vector count % = 0.8733624454148471\n"
     ]
    }
   ],
   "source": [
    "question_count = len(trainJSON[\"questions\"])\n",
    "question_embeddings = np.zeros((question_count, 1, embedding_dimension))\n",
    "question_labels = np.zeros((question_count, label_dim))\n",
    "questions_data = []\n",
    "for i, question in enumerate(trainJSON[\"questions\"]):\n",
    "    embedding = getQuestionembeddingsFromText(question[\"body\"])\n",
    "    question_embeddings[i:] = embedding\n",
    "    question_labels[i:] = labels[question[\"type\"]]\n",
    "    questions_data.append({\n",
    "        \"id\": question[\"id\"],\n",
    "        \"embedding\": embedding,\n",
    "        \"vector\": None\n",
    "    })\n",
    "    \n",
    "print(\"   \".join(no_vectors.keys()))\n",
    "n_cnt = len(no_vectors.keys())\n",
    "y_cnt = len(yes_vectors.keys())\n",
    "percent = ((n_cnt/(n_cnt+y_cnt)) * 100)\n",
    "print(\"\\n\\nNo vector count % = {}\".format(percent))\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(2251, 1, 200)\n",
      "(2251, 4)\n",
      "[[1. 0. 0. 0.]\n",
      " [0. 1. 0. 0.]\n",
      " [0. 0. 1. 0.]\n",
      " ...\n",
      " [0. 0. 0. 0.]\n",
      " [0. 0. 1. 0.]\n",
      " [0. 0. 1. 0.]]\n"
     ]
    }
   ],
   "source": [
    "print(question_embeddings.shape)\n",
    "print(question_labels.shape)\n",
    "print(question_labels[:-10])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [],
   "source": [
    "def getModel(input_shape):\n",
    "    X_input = keras.layers.Input(shape=input_shape)\n",
    "    X = keras.layers.Flatten()(X_input)\n",
    "    X = keras.layers.Dense(128, activation=tf.nn.relu)(X)\n",
    "    X = keras.layers.Dense(256, activation=tf.nn.relu)(X)\n",
    "    X = keras.layers.Dense(128, activation=tf.nn.relu)(X)\n",
    "    X = keras.layers.Dense(64, activation=tf.nn.relu)(X)\n",
    "    X = keras.layers.Dense(label_dim, activation=tf.nn.sigmoid)(X)\n",
    "    model = keras.Model(inputs=X_input, outputs=X, name='HappyModel')\n",
    "    return model"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [],
   "source": [
    "model = getModel((question_embeddings.shape[1], question_embeddings.shape[2]))\n",
    "model.compile(\n",
    "    optimizer=tf.train.AdamOptimizer(),\n",
    "    loss='binary_crossentropy',\n",
    "    metrics=['accuracy']\n",
    ")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "_________________________________________________________________\n",
      "Layer (type)                 Output Shape              Param #   \n",
      "=================================================================\n",
      "input_1 (InputLayer)         (None, 1, 200)            0         \n",
      "_________________________________________________________________\n",
      "flatten (Flatten)            (None, 200)               0         \n",
      "_________________________________________________________________\n",
      "dense (Dense)                (None, 128)               25728     \n",
      "_________________________________________________________________\n",
      "dense_1 (Dense)              (None, 256)               33024     \n",
      "_________________________________________________________________\n",
      "dense_2 (Dense)              (None, 128)               32896     \n",
      "_________________________________________________________________\n",
      "dense_3 (Dense)              (None, 64)                8256      \n",
      "_________________________________________________________________\n",
      "dense_4 (Dense)              (None, 4)                 260       \n",
      "=================================================================\n",
      "Total params: 100,164\n",
      "Trainable params: 100,164\n",
      "Non-trainable params: 0\n",
      "_________________________________________________________________\n"
     ]
    }
   ],
   "source": [
    "model.summary()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [],
   "source": [
    "x_train, x_test, y_train, y_test = train_test_split(question_embeddings, question_labels, test_size=0.2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Epoch 1/30\n",
      "1800/1800 [==============================] - 1s 544us/step - loss: 0.4466 - acc: 0.8136\n",
      "Epoch 2/30\n",
      "1800/1800 [==============================] - 0s 117us/step - loss: 0.3080 - acc: 0.8535\n",
      "Epoch 3/30\n",
      "1800/1800 [==============================] - 0s 113us/step - loss: 0.2180 - acc: 0.9062\n",
      "Epoch 4/30\n",
      "1800/1800 [==============================] - 0s 117us/step - loss: 0.2024 - acc: 0.9144\n",
      "Epoch 5/30\n",
      "1800/1800 [==============================] - 0s 112us/step - loss: 0.1718 - acc: 0.9308\n",
      "Epoch 6/30\n",
      "1800/1800 [==============================] - 0s 118us/step - loss: 0.1535 - acc: 0.9410\n",
      "Epoch 7/30\n",
      "1800/1800 [==============================] - 0s 116us/step - loss: 0.1476 - acc: 0.9446\n",
      "Epoch 8/30\n",
      "1800/1800 [==============================] - 0s 116us/step - loss: 0.1420 - acc: 0.9468\n",
      "Epoch 9/30\n",
      "1800/1800 [==============================] - 0s 120us/step - loss: 0.1247 - acc: 0.9546\n",
      "Epoch 10/30\n",
      "1800/1800 [==============================] - 0s 112us/step - loss: 0.1193 - acc: 0.9561\n",
      "Epoch 11/30\n",
      "1800/1800 [==============================] - 0s 110us/step - loss: 0.1097 - acc: 0.9597\n",
      "Epoch 12/30\n",
      "1800/1800 [==============================] - 0s 115us/step - loss: 0.1015 - acc: 0.9629\n",
      "Epoch 13/30\n",
      "1800/1800 [==============================] - 0s 109us/step - loss: 0.1032 - acc: 0.9608\n",
      "Epoch 14/30\n",
      "1800/1800 [==============================] - 0s 112us/step - loss: 0.1057 - acc: 0.9585\n",
      "Epoch 15/30\n",
      "1800/1800 [==============================] - 0s 112us/step - loss: 0.0875 - acc: 0.9681\n",
      "Epoch 16/30\n",
      "1800/1800 [==============================] - 0s 118us/step - loss: 0.0794 - acc: 0.9712\n",
      "Epoch 17/30\n",
      "1800/1800 [==============================] - 0s 116us/step - loss: 0.0760 - acc: 0.9718\n",
      "Epoch 18/30\n",
      "1800/1800 [==============================] - 0s 116us/step - loss: 0.0842 - acc: 0.9681\n",
      "Epoch 19/30\n",
      "1800/1800 [==============================] - 0s 111us/step - loss: 0.0650 - acc: 0.9760\n",
      "Epoch 20/30\n",
      "1800/1800 [==============================] - 0s 111us/step - loss: 0.0578 - acc: 0.9800\n",
      "Epoch 21/30\n",
      "1800/1800 [==============================] - 0s 110us/step - loss: 0.0493 - acc: 0.9840\n",
      "Epoch 22/30\n",
      "1800/1800 [==============================] - 0s 107us/step - loss: 0.0484 - acc: 0.9832\n",
      "Epoch 23/30\n",
      "1800/1800 [==============================] - 0s 116us/step - loss: 0.0381 - acc: 0.9868\n",
      "Epoch 24/30\n",
      "1800/1800 [==============================] - 0s 114us/step - loss: 0.0386 - acc: 0.9865\n",
      "Epoch 25/30\n",
      "1800/1800 [==============================] - 0s 114us/step - loss: 0.0480 - acc: 0.9810\n",
      "Epoch 26/30\n",
      "1800/1800 [==============================] - 0s 113us/step - loss: 0.0359 - acc: 0.9874\n",
      "Epoch 27/30\n",
      "1800/1800 [==============================] - 0s 109us/step - loss: 0.0293 - acc: 0.9903\n",
      "Epoch 28/30\n",
      "1800/1800 [==============================] - 0s 111us/step - loss: 0.0300 - acc: 0.9897\n",
      "Epoch 29/30\n",
      "1800/1800 [==============================] - 0s 109us/step - loss: 0.0232 - acc: 0.9933\n",
      "Epoch 30/30\n",
      "1800/1800 [==============================] - 0s 110us/step - loss: 0.0233 - acc: 0.9911\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "<tensorflow.python.keras.callbacks.History at 0x7fe443a6fac8>"
      ]
     },
     "execution_count": 16,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "model.fit(x=x_train, y=y_train, epochs=30)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "451/451 [==============================] - 0s 192us/step\n",
      "Loss = 0.31  and test accuracy = 0.93\n"
     ]
    }
   ],
   "source": [
    "test_loss, test_acc = model.evaluate(x_test, y_test)\n",
    "print('Loss = %.2f  and test accuracy = %.2f' % (test_loss, test_acc))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [],
   "source": [
    "def getPenultimateLayerOutput(model, input_embedding):\n",
    "    index = len(model.layers) - 2\n",
    "    get_2ndlast_layer_output = K.function([model.layers[0].input],\n",
    "                                      [model.layers[index].output])\n",
    "    layer_output = get_2ndlast_layer_output([input_embedding])[0]\n",
    "    return layer_output"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [],
   "source": [
    "question_vectors = getPenultimateLayerOutput(model, question_embeddings)\n",
    "for i, qVect in enumerate(question_vectors):\n",
    "    questions_data[i][\"vector\"] = qVect"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(2251, 64)"
      ]
     },
     "execution_count": 20,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "getPenultimateLayerOutput(model, question_embeddings).shape\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [],
   "source": [
    "def getQueryVector(qText):\n",
    "    qEmbedding = getQuestionembeddingsFromText(qText)\n",
    "    qVector = getPenultimateLayerOutput(model, qEmbedding)\n",
    "    return qVector"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [],
   "source": [
    "def getTop10MatchingQuestionIdsForQuery(qText):\n",
    "    qVector = getQueryVector(qText)\n",
    "    values = []\n",
    "    dtype = [('id', 'S50'), ('distance', float)]\n",
    "    for i, question in enumerate(questions_data):\n",
    "        values.append((\n",
    "            question[\"id\"],\n",
    "            cosine_similarity(qVector, question[\"vector\"].reshape((1, 64))).squeeze()\n",
    "        ))\n",
    "    distances = np.array(values, dtype=dtype)\n",
    "    questionid_cosine_distances = np.sort(distances, order=[\"distance\"])\n",
    "    return questionid_cosine_distances[-10:][::-1]\n",
    "    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 56,
   "metadata": {},
   "outputs": [],
   "source": [
    "def executeQuery(qText):\n",
    "    top_matches = getTop10MatchingQuestionIdsForQuery(qText)\n",
    "\n",
    "    for i, (id, dist) in enumerate(top_matches):\n",
    "        for question in trainJSON[\"questions\"]:\n",
    "            if id.decode(\"utf-8\") == question[\"id\"]:\n",
    "                print(\"\\nResult #%d\\n\"%(i+1))\n",
    "                print(\"------\")\n",
    "                print(question[\"ideal_answer\"])\n",
    "                print(\"\\n\\n\")\n",
    "                if \"snippets\" in question:\n",
    "                    for snippet in question[\"snippets\"]:\n",
    "                        print(snippet[\"text\"])\n",
    "                        print(\"\\n\")\n",
    "                    print(\"=\" * 100)\n",
    "                break"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 57,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Result #1\n",
      "\n",
      "------\n",
      "['Long non coding RNAs appear to be spliced through the same pathway as the mRNAs']\n",
      "\n",
      "\n",
      "\n",
      "Our analyses indicate that lncRNAs are generated through pathways similar to that of protein-coding genes, with similar histone-modification profiles, splicing signals, and exon/intron lengths.\n",
      "\n",
      "\n",
      "For alternative exons and long noncoding RNAs, splicing tends to occur later, and the latter might remain unspliced in some cases.\n",
      "\n",
      "\n",
      "bosome-mapping data to identify lncRNAs of Caenorhabditis elegans. We found 170 long intervening ncRNAs (lincRNAs), which had single- or multiexonic structures that did not overlap protein-coding transcripts, and about sixty antisense lncRNAs (ancRNAs), which were complementary to protein-coding transcripts\n",
      "\n",
      "\n",
      "We introduce an approach to predict spliced lncRNAs in vertebrate genomes combining comparative genomics and machine learning.\n",
      "\n",
      "\n",
      "Owing to similar alternative splicing pattern to mRNAs, the concept of lncRNA genes was put forward to help systematic understanding of lncRNAs. \n",
      "\n",
      "\n",
      "Our synthesis of recent studies suggests that neither size, presence of a poly-A tail, splicing, direction of transcription, nor strand specificity are of importance to lncRNA function.\n",
      "\n",
      "\n",
      "====================================================================================================\n",
      "\n",
      "Result #2\n",
      "\n",
      "------\n",
      "['No. Progesterone has been associated with robust positive effects in animal models of traumatic brain injury (TBI) and with clinical benefits in two phase 2 randomized, controlled trials. However, a recent large clinical trial showed no clinical benefit of progesterone in patients with severe TBI. These data stand in contrast to the robust preclinical data and results of early single-center trials that provided the impetus to initiate phase 3 trials.']\n",
      "\n",
      "\n",
      "\n",
      "BACKGROUND: Progesterone has been associated with robust positive effects in animal models of traumatic brain injury (TBI) and with clinical benefits in two phase 2 randomized, controlled trials. \n",
      "\n",
      "\n",
      "The proportion of patients with a favorable outcome on the Glasgow Outcome Scale (good recovery or moderate disability) was 50.4% with progesterone, as compared with 50.5% with placebo. Mortality was similar in the two groups. No relevant safety differences were noted between progesterone and placebo. CONCLUSIONS: Primary and secondary efficacy analyses showed no clinical benefit of progesterone in patients with severe TBI. These data stand in contrast to the robust preclinical data and results of early single-center trials that provided the impetus to initiate phase 3 trials.\n",
      "\n",
      "\n",
      "BACKGROUND: Traumatic brain injury (TBI) is a major cause of death and disability worldwide. Progesterone has been shown to improve neurologic outcome in multiple experimental models and two early-phase trials involving patients with TBI. \n",
      "\n",
      "\n",
      "There was no significant difference between the progesterone group and the placebo group in the proportion of patients with a favorable outcome (relative benefit of progesterone, 0.95; 95% confidence interval [CI], 0.85 to 1.06; P=0.35). Phlebitis or thrombophlebitis was more frequent in the progesterone group than in the placebo group (relative risk, 3.03; CI, 1.96 to 4.66). \n",
      "\n",
      "\n",
      "CONCLUSIONS: This clinical trial did not show a benefit of progesterone over placebo in the improvement of outcomes in patients with acute TBI. \n",
      "\n",
      "\n",
      "Numerous studies, however, show that progesterone has substantial pleiotropic properties as a neuroprotective agent in both animal models and humans.\n",
      "\n",
      "\n",
      "RESULTS: There was a better recovery rate and GOS score for the patients who were given progesterone than for those in the control group in a 3-months follow-up period (50% vs. 21%); subgroup analysis showed a significant difference in the percentage of favorable outcome between the two groups with GCS of 5-8 (p=0.03). CONCLUSION: The use of progesterone may significantly improve neurologic outcome of patients suffering severe TBI up to 3 months after injury, especially those with 5≤GCS≤8, providing a potential benefit to the treatment of acute severe TBI patients. Considering this drug had no significant side effects, so progesterone could be used in patients with severe TBI as a neuro-protective drug.\n",
      "\n",
      "\n",
      "While progesterone and ciclosporin have shown promise in phase II studies, success in larger phase III, randomized, multicentre, clinical trials is pending.\n",
      "\n",
      "\n",
      " All three studies reported the effects of progesterone on mortality. The pooled risk ratio (RR) for mortality at end of follow-up was 0.61, 95% confidence interval (CI) 0.40 to 0.93. Three studies measured disability and found the RR of death or severe disability in patients treated with progesterone to be 0.77, 95% CI 0.62 to 0.96.\n",
      "\n",
      "\n",
      "AUTHORS' CONCLUSIONS: Current clinical evidence from three small RCTs indicates progesterone may improve the neurologic outcome of patients suffering TBI. This evidence is still insufficient and further multicentre randomised controlled trials are required.\n",
      "\n",
      "\n",
      "GOS was classified to 2 main categories of favorable and unfavorable recovery, of which, favorable recovery in placebo, progesterone, and progesterone-vitamin D was 25%, 45%, and 60%, respectively which showed a statistical significant difference among the groups (P-value = 0.03). CONCLUSION: The results showed that recovery rate in patients with severe brain trauma in the group receiving progesterone and vitamin D together was significantly higher than that of progesterone group, which was in turn higher than that of placebo group.\n",
      "\n",
      "\n",
      " The pooled relative risk (RR) for mortality at end of follow-up is 0.61, 95% confidence interval (CI) 0.40 to 0.93. Three studies measured disability and found the RR of death or severe disability in patients treated with progesterone was 0.77, 95% confidence interval (CI) 0.62 to 0.96.\n",
      "\n",
      "\n",
      "AUTHORS' CONCLUSIONS: Current clinical evidence from three small RCTs indicates progesterone may improve the neurologic outcome of patients suffering TBI. This evidence is still insufficient and further multicentre randomised controlled trials are required.\n",
      "\n",
      "\n",
      "Clinical trials have shown that short-and long-term progesterone treatment induces a significant improvement in the level of disability among patients with brain injury. \n",
      "\n",
      "\n",
      "Improved outcomes from the administration of progesterone for patients with acute severe traumatic brain injury: a randomized controlled trial.\n",
      "\n",
      "\n",
      "CONCLUSION: Our data suggest that acute severe TBI patients with administration of progesterone hold improved neurologic outcomes for up to 6 months. These results provide information important for further large and multicenter clinical trials on progesterone as a promising neuroprotective drug. \n",
      "\n",
      "\n",
      "The modified Functional Independence Measure scores in the progesterone group were higher than those in the placebo group at both 3-month and 6-month follow-up (P < 0.05 and P < 0.01). The mortality rate of the progesterone group was significantly lower than that of the placebo group at 6-month follow-up (P < 0.05). \n",
      "\n",
      "\n",
      "CONCLUSION: Our data suggest that acute severe TBI patients with administration of progesterone hold improved neurologic outcomes for up to 6 months. These results provide information important for further large and multicenter clinical trials on progesterone as a promising neuroprotective drug. \n",
      "\n",
      "\n",
      "These data, combined with the results of the previously published ProTECT trial, show progesterone to be safe and potentially efficacious in the treatment of TBI. Larger phase III trials will be necessary to verify results prior to clinical implementation.\n",
      "\n",
      "\n",
      "CONCLUSION: It indicated that successive early application of PG will benefit the patients with acute severe head injury by improving the recovery and reducing the disability, which may be related to its alleviating inflammatory and lipid peroxidation response.\n",
      "\n",
      "\n",
      "Adverse and serious adverse event rates were similar in both groups, except that patients randomized to progesterone had a lower 30-day mortality rate than controls (rate ratio 0.43; 95% confidence interval 0.18 to 0.99). \n",
      "\n",
      "\n",
      "However, moderate traumatic brain injury survivors who received progesterone were more likely to have a moderate to good outcome than those randomized to placebo. CONCLUSION: In this small study, progesterone caused no discernible harm and showed possible signs of benefit.\n",
      "\n",
      "\n",
      "After more than 30 years of research and 30 failed clinical trials with as many different treatments, progesterone is the first agent to demonstrate robust clinical efficacy as a treatment for traumatic brain injuries.\n",
      "\n",
      "\n",
      "After more than 30 years of research and 30 failed clinical trials with as many different treatments, progesterone is the first agent to demonstrate robust clinical efficacy as a treatment for traumatic brain injuries\n",
      "\n",
      "\n",
      "A US National Institutes of Health-sponsored, nationwide Phase III clinical trial is now evaluating progesterone for moderate-to-severe TBI in 1200 patients\n",
      "\n",
      "\n",
      "An industry-sponsored Phase III international trial is also under way, and planning for a trial using progesterone to treat pediatric brain injury has begun\n",
      "\n",
      "\n",
      "More than two decades of pre-clinical research and two recent clinical trials have shown that progesterone (PROG) and its metabolites exert beneficial effects after traumatic brain injury (TBI) through a number of metabolic and physiological pathways that can reduce damage in many different tissues and organ systems\n",
      "\n",
      "\n",
      "After more than 30 years of research and 30 failed clinical trials with as many different treatments, progesterone is the first agent to demonstrate robust clinical efficacy as a treatment for traumatic brain injuries\n",
      "\n",
      "\n",
      "RESULTS: Analysis of these reviews yielded meanfuling observations: (1) The effectiveness of most ordinary treatments in TBI is inconclusive except that corticosteroids are likely to be ineffective or harmful, and tranexamic acid, nimodipine and progesterone show a promising effect in bleeding trauma, traumatic subarachnoid hemorrhage, TBI or severe TBI.\n",
      "\n",
      "\n",
      "Laboratory data strongly show that progesterone treatment after TBI reduces edema, improves outcomes, and restores blood-brain barrier function. Clinical studies to date agree with these data, and there are ongoing human trials for progesterone treatment after TBI.\n",
      "\n",
      "\n",
      "====================================================================================================\n",
      "\n",
      "Result #3\n",
      "\n",
      "------\n",
      "['Apixaban is a direct inhibitor of factor Xa, and is a potential alternative for the treatment of acute venous thromboembolism. These results suggest a lack of clear superiority of apixaban relative to enoxaparin. Apixaban is an oral alternative with similar efficacy and safety to existing anticoagulant therapies.', 'Yes, apixaban is effective for treatment of acute venous thromboembolismis. Apixiban is a direct inhibitor of factor Xa, and is a potential alternative for the treatment of acute venous thromboembolism. Apixaban is an oral alternative with similar efficacy and safety to existing anticoagulant therapies.']\n",
      "\n",
      "\n",
      "\n",
      "Apixaban is a direct inhibitor of factor Xa, and is a potential alternative for the treatment of acute venous thromboembolism. \n",
      "\n",
      "\n",
      "These results suggest a lack of clear superiority of apixaban relative to enoxaparin. Apixaban is an oral alternative with similar efficacy and safety to existing anticoagulant therapies.\n",
      "\n",
      "\n",
      "A fixed-dose regimen of apixaban alone was noninferior to conventional therapy for the treatment of acute venous thromboembolism and was associated with significantly less bleeding\n",
      "\n",
      "\n",
      "To critically review the effectiveness of the novel oral anticoagulants (rivaroxaban, dabigatran, ximelagatran, and apixaban) in the treatment of acute venous thromboembolism.\n",
      "\n",
      "\n",
      "ompared with vitamin K antagonists, the novel oral anticoagulants had a similar risk of recurrence of acute venous thromboembolism and all cause mortality, though rivaroxaban was associated with a reduced risk of bleeding\n",
      "\n",
      "\n",
      "Nowadays, the new anticoagulants, such as dabigatran, rivaroxaban and apixaban, show potential advantages over classical treatments. These agents inhibit specific coagulation factors and are administered orally at fixed doses.\n",
      "\n",
      "\n",
      "In a recently completed phase III trial, apixaban also demonstrated promising efficacy and safety in that indication\n",
      "\n",
      "\n",
      "the most advanced oral direct inhibitors to factor Xa (rivaroxaban and apixaban) and IIa (dabigatran)\n",
      "\n",
      "\n",
      "====================================================================================================\n",
      "\n",
      "Result #4\n",
      "\n",
      "------\n",
      "[\"Dysregulation of miRNA expression involved in cancer and Alzheimer's disease can be triggered by multiple mechanisms including aberrant DNA methylation of the miRNA gene promoter. Epigenetic dysregulation of tumor-suppressor miRNA genes by promoter DNA methylation has been implicated in human cancers, including multiple myeloma (MM).\", 'Dysregulation of miRNA expression involved in cancer can be triggered by multiple mechanisms including aberrant DNA methylation of the miRNA gene promoterRecently, epigenetic dysregulation of tumor-suppressor miRNA genes by promoter DNA methylation has been implicated in human cancers, including multiple myeloma (MM)']\n",
      "\n",
      "\n",
      "\n",
      "We found that Tcf3 down-regulation in the context of constitutively active Wnt signaling does not result from promoter DNA methylation but is likely to be caused by a plethora of mechanisms at both the RNA and protein level as shown by the observed decrease in activating histone marks (H3K4me3 and H3-acetylation) and the upregulation of miR-211, a novel Wnt-regulated microRNA that targets Tcf3 and attenuates early neural differentiation in mouse ESCs\n",
      "\n",
      "\n",
      "ene silencing of MIR22 in acute lymphoblastic leukaemia involves histone modifications independent of promoter DNA methylation\n",
      "\n",
      "\n",
      "Whereas a CpG island was identified within the promoter element of MIR22, no promoter DNA methylation was detected in these cells\n",
      "\n",
      "\n",
      "xtensive promoter DNA hypermethylation and hypomethylation is associated with aberrant microRNA expression in chronic lymphocytic leukemia\n",
      "\n",
      "\n",
      "Integration of DNA methylation and miRNA promoter data led to the identification of 128 recurrent miRNA targets for aberrant promoter DNA methylation\n",
      "\n",
      "\n",
      "Together, our findings characterize the role of epigenetic changes in the regulation of miRNA transcription and create a repository of disease-specific promoter regions that may provide additional insights into the pathogenesis of CLL\n",
      "\n",
      "\n",
      "NA methylation of microRNA genes in multiple myeloma\n",
      "\n",
      "\n",
      "Recently, epigenetic dysregulation of tumor-suppressor miRNA genes by promoter DNA methylation has been implicated in human cancers, including multiple myeloma (MM)\n",
      "\n",
      "\n",
      "ethylation of tumor suppressor microRNAs\n",
      "\n",
      "\n",
      "Dysregulation of miRNA expression involved in cancer can be triggered by multiple mechanisms including aberrant DNA methylation of the miRNA gene promoter\n",
      "\n",
      "\n",
      "Of note, DNA methylation of tumor suppressor miRNAs has been implicated in various human cancers\n",
      "\n",
      "\n",
      "Moreover, miRNA silencing mediated by aberrant promoter DNA methylation can potentially be reversed by hypomethylating agents, and hence may pose a new therapeutic target in cancer\n",
      "\n",
      "\n",
      " In this review, the authors will focus on the aberrant methylation of miRNAs in the pathogenesis of lymphoid malignancies including chronic lymphocytic leukemia, multiple myeloma and acute lymphoblastic leukemia\n",
      "\n",
      "\n",
      "Here, we review those miRNAs implicated in AD that are regulated by promoter DNA methylation and/or chromatin modifications and, which frequently direct the expression of constituents of the epigenetic machinery, concluding with the delineation of a complex epigenetic-miRNA regulatory network and its alterations in AD\n",
      "\n",
      "\n",
      "Furthermore, we also discuss epigenetic dysregulation of tumor-suppressor miRNA genes by promoter DNA methylation and the interaction of DNA methylation with miRNAs involved in the regulation of HSC activation and liver fibrosi\n",
      "\n",
      "\n",
      "Instead, the cell type-specific silencing of these genes is due to enhanced p21 mRNA degradation, 14-3-3sigma promoter DNA methylation and reduced processing of the miR-34a primary transcript\n",
      "\n",
      "\n",
      "Some tumor-suppressive miRNAs are known to be epigenetically silenced by promoter DNA methylation in cancer\n",
      "\n",
      "\n",
      "In the present study, we aimed to identify miRNA genes that are silenced by DNA hypermethylation in hepatocellular carcinoma (HCC)\n",
      "\n",
      "\n",
      "It was found that miR-335, which is harbored within an intron of its protein-coding host gene, MEST, was downregulated by aberrant promoter hypermethylation via further methylation assays, including methylation-specific PCR, combined bisulfite and restriction analysis, bisulfite sequencing analysis and 5-aza-2'-deoxycytidine treatment\n",
      "\n",
      "\n",
      "he levels of miR-335/MEST methylation were significantly higher in 18 (90%) out of 20 primary HCC tumors, compared to their non-tumor tissue counterparts (P<0.001)\n",
      "\n",
      "\n",
      "In conclusion, our results indicate that expression of miR-335 is reduced by aberrant DNA methylation in HCC.\n",
      "\n",
      "\n",
      "====================================================================================================\n",
      "\n",
      "Result #5\n",
      "\n",
      "------\n",
      "[\"Our analysis indicates that the association of CGIs with housekeeping genes is not as strong as previously estimated. These regions represent about 1% of genomic DNA and are generally found in the promoter region of housekeeping genes. In housekeeping and many tissue-specific genes, the promoter is embedded in a so-called CpG island. Methylation-free CpG clusters, so-called HTF islands, are most often associated with the promoter regions of housekeeping genes, whereas genes expressed in a single-cell type are usually deficient in these sequences. All housekeeping and widely expressed genes have a CpG island covering the transcription start, whereas 40% of the genes with a tissue-specific or limited expression are associated with islands. It has been envisaged that CpG islands are often observed near the transcriptional start sites (TSS) of housekeeping genes. CpG islands, which are found almost exclusively at the 5'-end of housekeeping genes. CpG islands were associated with the 5' ends of all housekeeping genes and many tissue-specific genes, and with the 3' ends of some tissue-specific genes. \", 'Yes, CpG islands are preferentially located at the start of transcription of housekeeping genes and are associated with tissue-specific genes.', \"These regions represent about 1% of genomic DNA and are generally found in the promoter region of housekeeping genes. Our analysis indicates that the association of CGIs with housekeeping genes is not as strong as previously estimated. In housekeeping and many tissue-specific genes, the promoter is embedded in a so-called CpG island. Methylation-free CpG clusters, so-called HTF islands, are most often associated with the promoter regions of housekeeping genes, whereas genes expressed in a single-cell type are usually deficient in these sequences. All housekeeping and widely expressed genes have a CpG island covering the transcription start, whereas 40% of the genes with a tissue-specific or limited expression are associated with islands. It has been envisaged that CpG islands are often observed near the transcriptional start sites (TSS) of housekeeping genes. CpG islands, which are found almost exclusively at the 5'-end of housekeeping genes. CpG islands were associated with the 5' ends of all housekeeping genes and many tissue-specific genes, and with the 3' ends of some tissue-specific genes. \"]\n",
      "\n",
      "\n",
      "\n",
      "our analysis indicates that the association of CGIs with housekeeping genes is not as strong as previously estimated\n",
      "\n",
      "\n",
      "CpG islands are preferentially located at the start of transcription of housekeeping genes and are associated with tissue-specific genes\n",
      "\n",
      "\n",
      "It has been envisaged that CpG islands are often observed near the transcriptional start sites (TSS) of housekeeping genes.\n",
      "\n",
      "\n",
      "These regions represent about 1% of genomic DNA and are generally found in the promoter region of housekeeping genes.\n",
      "\n",
      "\n",
      "CpG islands are stretches of DNA sequence that are enriched in the (CpG)n repeat and are present in close association with all housekeeping genes as well as some tissue-specific genes in the mammalian genome.\n",
      "\n",
      "\n",
      "CpG islands, which are found almost exclusively at the 5'-end of housekeeping genes\n",
      "\n",
      "\n",
      "In housekeeping and many tissue-specific genes, the promoter is embedded in a so-called CpG island.\n",
      "\n",
      "\n",
      "All housekeeping and widely expressed genes have a CpG island covering the transcription start, whereas 40% of the genes with a tissue-specific or limited expression are associated with islands\n",
      "\n",
      "\n",
      "Methylation-free CpG clusters, so-called HTF islands, are most often associated with the promoter regions of housekeeping genes, whereas genes expressed in a single-cell type are usually deficient in these sequences.\n",
      "\n",
      "\n",
      "Unmethylated CpG rich islands are a feature of vertebrate DNA: they are associated with housekeeping and many tissue specific genes.\n",
      "\n",
      "\n",
      "CpG islands were associated with the 5' ends of all housekeeping genes and many tissue-specific genes, and with the 3' ends of some tissue-specific genes.\n",
      "\n",
      "\n",
      "====================================================================================================\n",
      "\n",
      "Result #6\n",
      "\n",
      "------\n",
      "['Statin use after subarachnoid hemorrhage has been shown be associated with improved outcomes by some prospective clinical trials. It has been reported that statin use after subarachnoid hemorrhage reduced rates of vasospasm, delayed cerebral ischemia, and mortality. However, other authors have failed to find beneficial effect of statin use in subarachnoid hemorrhage patients.']\n",
      "\n",
      "\n",
      "\n",
      "Statins have been shown in two recent small phase I/II trials to be associated with a marked reduction in clinical and transcranial Doppler (TCD) evidence of vasospasm after aneurysmal subarachnoid haemorrhage (SAH). \n",
      "\n",
      "\n",
      "Statins did not result in reduced TCD velocities, clinical or angiographic vasospasm, or improvements in global outcome at the time of hospital discharge. \n",
      "\n",
      "\n",
      "There remains significant uncertainty as to the role of statins in preventing vasospasm after SAH.\n",
      "\n",
      "\n",
      "Although the results of 2 randomized clinical trials demonstrated that statin decreases the incidence of symptomatic cerebral vasospasm after aSAH, retrospective studies have failed to confirm this.\n",
      "\n",
      "\n",
      "There were no differences in the incidence of symptomatic vasospasm (25.3 vs 30.5%; p = 0.277), in-hospital mortality rate (18 vs 15%; p = 0.468), length of hospitalization (21 +/- 15 vs 19 +/- 12 days; p = 0.281), or poor outcome at discharge (Glasgow Outcome Scale Scores 1-2: 21.7 vs 18.2%; p = 0.416) between the simvastatin and nonstatin cohorts. \n",
      "\n",
      "\n",
      "The uniform introduction of simvastatin did not reduce the incidence of symptomatic cerebral vasospasm, death, or poor outcome in patients with aSAH. Simvastatin was well tolerated, but its benefit may be less than has been previously reported.\n",
      "\n",
      "\n",
      "Cholesterol-reducing agents might improve unfavourable outcomes.\n",
      "\n",
      "\n",
      "We cannot draw any conclusions about the effectiveness and safety of lowering cholesterol in aneurysmal SAH because of insufficient reliable evidence from only one small trial.\n",
      "\n",
      "\n",
      "Experimental evidence has indicated the benefit of simvastatin in the treatment of subarachnoid hemorrhage.\n",
      "\n",
      "\n",
      "There was an improvement in the functional outcome in the simvastatin group at 1, 3 or 6 months in the follow-up; however, this difference was not statistically significant.\n",
      "\n",
      "\n",
      " There was benefit of simvastatin in terms of reduction in clinical vasospasm, mortality or improved functional outcome, however, this was not statistically significant.\n",
      "\n",
      "\n",
      "Cerebral vasomotor reactivity, however, is significantly improved after long-term statin administration in most patients with severe small vessel disease, aneurysmal subarachnoid hemorrhage, or impaired baseline CA.\n",
      "\n",
      "\n",
      "Atorvastatin decreases computed tomography and S100-assessed brain ischemia after subarachnoid aneurysmal hemorrhage: a comparative study.\n",
      "\n",
      "\n",
      "In the overall population, cerebral vasospasm was significantly less common in the statin-treated group. Severity of vasospasm, as assessed on the most severe angiogram, was lowered with statin. Statins significantly reduced volume of ischemia in patients with vasospasm and an uncomplicated coiling procedure. S100B levels were significantly lower in statin-treated patients, and the decrease was greatest among high-grade patients (World Federation of Neurological Surgeons 3-5). No differences were found between statin-treated and untreated groups regarding rescue therapy intensity or 1-yr clinical outcomes.\n",
      "\n",
      "\n",
      "Atorvastatin reduces the incidence, the severity and the ischemic consequences of vasospasm as assessed on computed tomography. In high-grade World Federation of Neurological Surgeons patients, atorvastatin decreases serum levels of S100B, a biomarker of brain ischemia. Despite these positive effects on biomarkers, no improvement of outcome was seen in the overall population, although there was a tendency for a better clinical outcome in high-grade patients.\n",
      "\n",
      "\n",
      "3-hydroxy-3-methylglutaryl coenzyme A (HMG-CoA) reductase inhibitors, or statins, have been associated with improved clinical outcomes after ischemic stroke and subarachnoid hemorrhage, but with an increased risk of incidental spontaneous intracerebral hemorrhage (ICH).\n",
      "\n",
      "\n",
      "Statins are known to have pleiotropic vascular effects, some of which may interrupt the pathogenesis of DNDs. Based on promising preliminary reports, many clinicians routinely administer statins to prevent DNDs.\n",
      "\n",
      "\n",
      "However, observational studies have not revealed an association between statin-use and reduced DNDs or improved neurological outcomes. Results of RCTs have been inconsistent and limited by small sample size, but together suggest that statins may reduce DNDs, with no clear impact on mortality or neurological recovery.\n",
      "\n",
      "\n",
      "the role of statins in the management of patients with SAH remains unclear. Although promising, statins should not, at this time, be considered standard care.\n",
      "\n",
      "\n",
      "In patients with SAH, they may decrease the incidence of symptomatic vasospasm, although the effects on overall outcome are less clear.\n",
      "\n",
      "\n",
      "Statins treatment may have potential clinical impact in vascular disease beyond cholesterol lowering. Its benefits have been documented in cerebral ischaemia and in subarachnoid haemorrhage.\n",
      "\n",
      "\n",
      "A recent meta-analysis investigating the efficacy of statin treatment in patients with aneurysmal subarachnoid hemorrhage reported a reduced incidence of vasospasm, delayed cerebral ischemia, and mortality in statin-treated patients.\n",
      "\n",
      "\n",
      "The results of the present systematic review do not lend statistically significant support to the finding of a beneficial effect of statins in patients with aneurysmal subarachnoid hemorrhage as reported in a previous meta-analysis.\n",
      "\n",
      "\n",
      "Pre-treatment with cholesterol lowering drugs of the statin family may exert protective effects in patients with ischaemic stroke and subarachnoid haemorrhage but their effects are not clear in patients with intracerebral haemorrhage (ICH). \n",
      "\n",
      "\n",
      "Recently, two randomized controlled phase II studies showed that acute initiation of statin treatment directly after aneurysmal subarachnoid hemorrhage (SAH) decreases the incidence of radiologic vasospasm and clinical signs of delayed cerebral ischemia (DCI), and even reduces mortality.\n",
      "\n",
      "\n",
      "We conclude that both the primary and secondary outcome results of this study do not support a beneficial effect of simvastatin in patients with SAH.\n",
      "\n",
      "\n",
      "Novel uses of their anti-inflammatory properties in sepsis and vasomotor properties in subarachnoid haemorrhage are being further investigated by randomised trials.\n",
      "\n",
      "\n",
      "A trend towards a lower mortality within 14 days in patients receiving solely simvastatin and those receiving statin and magnesium as compared with the control group was found. \n",
      "\n",
      "\n",
      "Initiation of statin therapy after aneurysmal SAH significantly reduces the incidence of vasospasm, delayed ischemic deficits, and mortality.\n",
      "\n",
      "\n",
      "The addition of statins to standard care was not associated with any reduction in the development of vasospasm or improvement in outcomes after aneurysmal subarachnoid hemorrhage. \n",
      "\n",
      "\n",
      "We have previously demonstrated that acute pravastatin therapy after aneurysmal subarachnoid hemorrhage ameliorates vasospasm-related delayed ischemic deficits.\n",
      "\n",
      "\n",
      " This trial demonstrates that acute statin treatment reduces traditional rescue therapy for vasospasm after aneurysmal subarachnoid hemorrhage. Improvement in early outcome has proved robust at 6 months, particularly in relation to physical and psychosocial (Short Form 36) outcome.\n",
      "\n",
      "\n",
      "The authors previously have demonstrated that acute treatment with pravastatin after aneurysmal subarachnoid hemorrhage (SAH) can ameliorate vasospasm-related delayed ischemic neurological deficits (DINDs).\n",
      "\n",
      "\n",
      "The neuroprotective effects of acute treatment with pravastatin following aneurysmal SAH are associated with enhancement of autoregulation\n",
      "\n",
      "\n",
      "Simvastatin reduces vasospasm after aneurysmal subarachnoid hemorrhage: results of a pilot randomized clinical trial.\n",
      "\n",
      "\n",
      "The use of simvastatin as prophylaxis against delayed cerebral ischemia after aneurysmal SAH is a safe and well-tolerated intervention. Its use attenuates serum markers associated with brain injury and decreases the incidence of radiographic vasospasm and delayed ischemic deficit.\n",
      "\n",
      "\n",
      "Acute treatment with pravastatin after aSAH is safe and ameliorates cerebral vasospasm, improves cerebral autoregulation, and reduces vasospasm-related DID.\n",
      "\n",
      "\n",
      " SAH statin users demonstrated significant improvement in 14-day functional outcome, a significantly lower incidence of DCI and cerebral infarctions of any type, as well as prevention of TCD highest mean velocity elevation. However, we did not find a significant statin impact on mortality or global outcome (Modified Rankin Scale) in this small sample. \n",
      "\n",
      "\n",
      "====================================================================================================\n",
      "\n",
      "Result #7\n",
      "\n",
      "------\n",
      "[\"Yes, platelet biomarkers can be used to study Alzheimer's disease.\"]\n",
      "\n",
      "\n",
      "\n",
      "Platelet biomarkers in Alzheimer's disease.\n",
      "\n",
      "\n",
      "platelets are the most important source of circulating forms of the amyloid precursor protein and other important proteins such as Tau and glycogen synthase kinase-3B.\n",
      "\n",
      "\n",
      "Alternative plasma and platelet measures are described,\n",
      "\n",
      "\n",
      "The success of these studies led to the application of platelet proteomics to the study of several pathologies where platelets play a fundamental role. Those include platelet-related disorders, such as storage pool disease, gray platelet syndrome, and Quebec platelet disorder; diseases where unwanted platelet activation is highly relevant, such as thrombosis and cardiovascular disease; and other diseases, such as cystic fibrosis, uremia, or Alzheimer's disease. \n",
      "\n",
      "\n",
      "====================================================================================================\n",
      "\n",
      "Result #8\n",
      "\n",
      "------\n",
      "['Yes, there is evidence to suggest that regorafenib can be effective for sarcoma treatment. Clinical trials are under-way.']\n",
      "\n",
      "\n",
      "\n",
      "Regorafenib has been approved for third-line therapy.\n",
      "\n",
      "\n",
      "Study protocol of REGOSARC trial: activity and safety of regorafenib in advanced soft tissue sarcoma: a multinational, randomized, placebo-controlled, phase II trial.\n",
      "\n",
      "\n",
      "DISCUSSION: The design of this trial allows an assessment of regorafenib activity over placebo in four sarcoma strata and might provide evidence for launching a phase III trial.\n",
      "\n",
      "\n",
      "This case provides rationale for adding a Ewing sarcoma arm to SARC024, a phase II study of regorafenib, another multi-targeted kinase inhibitor, in patients with liposarcoma, osteosarcoma and Ewing and Ewing-like sarcomas (NCT02048371).\n",
      "\n",
      "\n",
      "Thus, the Phase III studies with pazopanib, regorafenib, muramyl tripeptide (MTP) and ridaforolimus are extensively discussed as well as the biological rationale for the use of these compounds.\n",
      "\n",
      "\n",
      "Currently, regorafenib is examined in several clinical trials (mostly phase II) in different tumor entities, including renal cell carcinoma (RCC), hepatocellular carcinoma (HCC), and soft tissue sarcoma (STS).\n",
      "\n",
      "\n",
      "Analysis of primary human sarcoma samples revealed direct cytotoxicity following exposure to sorafenib and regorafenib with a corresponding increase in ALDHbright cells (P<0.05).\n",
      "\n",
      "\n",
      "Parametric and non-parametric statistical analyses were performed as appropriate.RESULTS: After functionally validating the CSC phenotype of ALDHbright sarcoma cells, we observed that sorafenib and regorafenib were cytotoxic to sarcoma cell lines (P<0.05), with a corresponding 1.4 - 2.8 fold increase in ALDHbright cells from baseline (P<0.05).\n",
      "\n",
      "\n",
      "We evaluated survival and CSC phenotype in mice harboring sarcoma metastases after TKI therapy. We exposed dissociated primary sarcoma tumors to sorafenib, regorafenib, and pazopanib, and we used tissue microarray (TMA) and primary sarcoma samples to evaluate the frequency and intensity of CSC markers after neoadjuvant therapy with sorafenib and pazopanib. Parametric and non-parametric statistical analyses were performed as appropriate.RESULTS: After functionally validating the CSC phenotype of ALDHbright sarcoma cells, we observed that sorafenib and regorafenib were cytotoxic to sarcoma cell lines (P<0.05), with a corresponding 1.4 - 2.8 fold increase in ALDHbright cells from baseline (P<0.05).\n",
      "\n",
      "\n",
      "We exposed dissociated primary sarcoma tumors to sorafenib, regorafenib, and pazopanib, and we used tissue microarray (TMA) and primary sarcoma samples to evaluate the frequency and intensity of CSC markers after neoadjuvant therapy with sorafenib and pazopanib. Parametric and non-parametric statistical analyses were performed as appropriate.RESULTS: After functionally validating the CSC phenotype of ALDHbright sarcoma cells, we observed that sorafenib and regorafenib were cytotoxic to sarcoma cell lines (P<0.05), with a corresponding 1.4 - 2.8 fold increase in ALDHbright cells from baseline (P<0.05).\n",
      "\n",
      "\n",
      "We exposed dissociated primary sarcoma tumors to sorafenib, regorafenib, and pazopanib, and we used tissue microarray (TMA) and primary sarcoma samples to evaluate the frequency and intensity of CSC markers after neoadjuvant therapy with sorafenib and pazopanib. Parametric and non-parametric statistical analyses were performed as appropriate.RESULTS: After functionally validating the CSC phenotype of ALDHbright sarcoma cells, we observed that sorafenib and regorafenib were cytotoxic to sarcoma cell lines (P<0.05), with a corresponding 1.4 - 2.8 fold increase in ALDHbright cells from baseline (P<0.05). In contrast, we observed negligible effects on viability and CSC sub-populations with pazopanib.\n",
      "\n",
      "\n",
      "After functionally validating the CSC phenotype of ALDHbright sarcoma cells, we observed that sorafenib and regorafenib were cytotoxic to sarcoma cell lines (P < 0.05), with a corresponding 1.4 - 2.8 fold increase in ALDHbright cells from baseline (P < 0.05).\n",
      "\n",
      "\n",
      "====================================================================================================\n",
      "\n",
      "Result #9\n",
      "\n",
      "------\n",
      "['FOXOs are reliable markers of longevity.']\n",
      "\n",
      "\n",
      "\n",
      "Forkhead box O (FOXO) transcription factors have a conserved function in regulating metazoan lifespan.\n",
      "\n",
      "\n",
      "In contrast to FoxO1, FoxO3a and FoxO6 were specifically diminished in the CNS of HFD animals possibly contributing to the reduced lifespan observed in these animals.\n",
      "\n",
      "\n",
      "Interestingly, many target proteins of AMPK are so-called longevity factors, e.g., SIRT1, p53, and FoxOs, which not only can increase the stress resistance and extend the lifespan of many organisms but also inhibit the inflammatory responses. \n",
      "\n",
      "\n",
      "Components of anti-ageing and autophagy include SirTs and FoxOs.\n",
      "\n",
      "\n",
      "Since Sirts and FoxOs are reliable markers of longevity, the results appear to suggest that Longevinex induces longevity after prolonged feeding via induction of autophagy, while it converts death signals into survival signals and provides cardioprotection within a relatively shorter period of time.\n",
      "\n",
      "\n",
      "Forkhead box O (FOXO) transcription factors are involved in various cellular processes, including cell proliferation, stress resistance, metabolism, and longevity\n",
      "\n",
      "\n",
      "In this respect, members of the mammalian forkhead transcription factors of the O class (FoxOs) that include FoxO1, FoxO3, FoxO4 and FoxO6 are increasingly being recognized as exciting prospects for multiple disorders. These transcription factors govern development, proliferation, survival and longevity during multiple cellular environments that can involve oxidative stress. \n",
      "\n",
      "\n",
      "Here we discuss the fascinating but complex role of FoxOs during cellular injury and oxidative stress, progenitor cell development, fertility, angiogenesis, cardiovascular function, cellular metabolism and diabetes, cell longevity, immune surveillance and cancer.\n",
      "\n",
      "\n",
      "Many longevity genes, e.g. FoxOs and SIRT1, are inhibitors of NF-kappaB signaling. \n",
      "\n",
      "\n",
      "Interestingly, several longevity genes such as SIRT1, SIRT6, and FoxOs can clearly suppress NF-kappaB signaling and in this way delay the aging process and extend lifespan.\n",
      "\n",
      "\n",
      "Yet, FoxOs also can significantly affect normal cell survival and longevity, requiring new treatments for neoplastic growth to modulate novel pathways that integrate cell proliferation, metabolism, inflammation and survival.\n",
      "\n",
      "\n",
      "These observations link FoxO function in mammalian systems with the evolutionarily conserved role of FoxO in promotion of stress resistance and longevity in lower phylogenetic systems. Furthermore, these findings have implications for aging in higher organisms and in malignant stem cell biology, and suggest that FoxOs may play an important role in the maintenance and integrity of stem cell compartments in a broad spectrum of tissues.\n",
      "\n",
      "\n",
      "Forkhead box O (FoxO) transcription factors are important downstream targets of the PI3K/Akt signaling pathway and crucial regulators of cell fate. This function of FoxOs relies on their ability to control diverse cellular functions, including proliferation, differentiation, apoptosis, DNA repair, defense against oxidative stress and ageing.\n",
      "\n",
      "\n",
      "This brief review focuses on the molecular mechanisms, cellular effects and resulting organismal phenotypes generated by differentially regulated FoxO proteins and discusses our current understanding of the role of FoxOs in disease and ageing processes.\n",
      "\n",
      "\n",
      "In this review, we focus on the several interactions of aging-associated signaling cascades regulated either by Sirtuins and FoxOs or NF-kappaB signaling pathways. We provide evidence that signaling via the longevity factors of FoxOs and SIRT1 can inhibit NF-kappaB signaling and simultaneously protect against inflamm-aging process.\n",
      "\n",
      "\n",
      "In diverse species transcription factors belonging to the forkhead/winged helix box gene, group O (FOXO) subfamily have been found to be crucial in downstream suppression of the life-shortening effects of insulin/insulin-like growth factor-I receptor signalling pathways that, when upregulated, accelerate ageing by suppression of FOXO. \n",
      "\n",
      "\n",
      "In humans, FOXO3a, as well as FOXO1 and -4, and their downstream effectors, could hold the key to counteracting ageing and common diseases.\n",
      "\n",
      "\n",
      "FOXO transcription factors have important roles in metabolism, cellular proliferation, stress tolerance, and aging. \n",
      "\n",
      "\n",
      "====================================================================================================\n",
      "\n",
      "Result #10\n",
      "\n",
      "------\n",
      "['Yes. Deep learning has been used so far in genomics for predicting splicing patterns in individual tissues and differences in splicing patterns across tissues. The deep architecture surpasses the performance of the previous Bayesian method for predicting alternative splicing (AS) patterns.']\n",
      "\n",
      "\n",
      "\n",
      "Deep learning of the tissue-regulated splicing code.\n",
      "\n",
      "\n",
      "Using a deep neural network, we developed a model inferred from mouse RNA-Seq data that can predict splicing patterns in individual tissues and differences in splicing patterns across tissues. Our architecture uses hidden variables that jointly represent features in genomic sequences and tissue types when making predictions. A graphics processing unit was used to greatly reduce the training time of our models with millions of parameters.\n",
      "\n",
      "\n",
      "We show that the deep architecture surpasses the performance of the previous Bayesian method for predicting AS patterns. With the proper optimization procedure and selection of hyperparameters, we demonstrate that deep architectures can be beneficial, even with a moderately sparse dataset. An analysis of what the model has learned in terms of the genomic features is presented.\n",
      "\n",
      "\n",
      "Machine learning applications in genetics and genomics\n",
      "\n",
      "\n",
      "====================================================================================================\n"
     ]
    }
   ],
   "source": [
    "executeQuery(\"Are long non coding RNAs spliced?\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 58,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Result #1\n",
      "\n",
      "------\n",
      "['Receptor activator of nuclear factor κB ligand (RANKL) is a cytokine predominantly secreted by osteoblasts.']\n",
      "\n",
      "\n",
      "\n",
      "Osteoprotegerin (OPG) is a soluble secreted factor that acts as a decoy receptor for receptor activator of NF-κB ligand (RANKL) \n",
      "\n",
      "\n",
      "Osteoprotegerin (OPG) is a secreted glycoprotein and a member of the tumor necrosis factor receptor superfamily. It usually functions in bone remodeling, by inhibiting osteoclastogenesis through interaction with a receptor activator of the nuclear factor κB (RANKL).\n",
      "\n",
      "\n",
      "e RANKL/OPG ratio secreted by osteoblasts increased and RANK expression by osteoclasts increased, leading to increased osteoclastogenesis\n",
      "\n",
      "\n",
      "Osteoprotegerin (OPG) is an essential secreted protein in bone turnover due to its role as a decoy receptor for the Receptor Activator of Nuclear Factor-kB ligand (RANKL) in the osteoclasts, thus inhibiting their differentiation\n",
      "\n",
      "\n",
      "We identify a TNFSF11 transcript variant that extends the originally identified transcript encoding secreted RANKL.\n",
      "\n",
      "\n",
      "Activated human T cells express alternative mRNA transcripts encoding a secreted form of RANKL.\n",
      "\n",
      "\n",
      "OPG, on the other hand, is secreted by osteoblast as a decoy receptor for RANKL, prevents RANKL from binding to RANK and thus prevents bone resorption\n",
      "\n",
      "\n",
      "Receptor activator of nuclear factor κB ligand (RANKL) and osteoprotegerin (OPG) are cytokines predominantly secreted by osteoblasts and play a central role in differentiation and functional activation of osteoclasts\n",
      "\n",
      "\n",
      "Although B. abortus-activated T cells actively secreted the pro-osteoclastogenic cytokines RANKL and IL-17, osteoclastogenesis depended on IL-17, because osteoclast generation induced by Brucella-activated T cells was completely abrogated when these cells were cultured with BMMs from IL-17 receptor knockout mice. \n",
      "\n",
      "\n",
      " osteoclastogenesis and bone destruction in autoimmune arthritis. We isolated human fibroblasts from RA, pyrophosphate arthropathy (PPA) and osteoarthritis (OA) patients and analyzed their RANKL/OPG expression profile and the capacity of their secreted factors to induce osteoclastogenesis.\n",
      "\n",
      "\n",
      "Osteoprotegerin (OPG) and receptor activator of nuclear factor κB ligand (RANKL) are cytokines predominantly secreted by osteoblasts and play critical roles in the differentiation and function of osteoclasts. \n",
      "\n",
      "\n",
      "====================================================================================================\n",
      "\n",
      "Result #2\n",
      "\n",
      "------\n",
      "['Yes, the gorilla genome has been sequenced.']\n",
      "\n",
      "\n",
      "\n",
      "Starting with human, chimpanzee, gorilla, and orangutan genomes, our software generated an exhaustive data set of 292 ALs (∼1 kb each) in ∼3 h. \n",
      "\n",
      "\n",
      "We generated a high-quality assembly of the gorilla genome using single-molecule, real-time sequence technology and a string graph de novo assembly algorithm\n",
      "\n",
      "\n",
      "Here we present the assembly and analysis of a genome sequence for the western lowland gorilla, and compare the whole genomes of all extant great ape genera.\n",
      "\n",
      "\n",
      "DNA sequencing reveals that the genomes of the human, gorilla and chimpanzee share more than 98% homology.\n",
      "\n",
      "\n",
      "====================================================================================================\n",
      "\n",
      "Result #3\n",
      "\n",
      "------\n",
      "['Yes, induction of alpha-tubulin acetylation correlates with inhibition of cell motility, while it is involved in additional cellular processes, e.g. cell cycle progression, differentiation, intracellular trafficking, and signalling. Dynamic microtubule (MT) acetylation/deacetylation mediating cell motility and adhesion is controlled by enzymes such as HDAC6, a major cytoplasmic α-tubulin deacetylase. While its overexpression and activation is capable to enhance cell motility, HDAC6 activity can also be negatively regulated by a number of cellular inhibitors, thus decreasing the ability of cells for migration.']\n",
      "\n",
      "\n",
      "\n",
      "In this study, we found that paclitaxel induced tubulin acetylation in endothelial and tumor cells, at concentrations that affected cell motility but not proliferation (10(-8) to 10(-9) M, for 4 hours). Induction of tubulin acetylation correlated with inhibition of motility but not proliferation based on a comparison of highly and poorly cytotoxic taxanes (paclitaxel and IDN5390) and tumor cell lines sensitive and resistant to paclitaxel (1A9 and 1A9 PTX22).\n",
      "\n",
      "\n",
      "we found that overexpression of the tubulin deacetylase SIRT2 increased cell motility and reduced cell response to the anti-motility activity of paclitaxel. Conversely, the SIRT2 inhibitor splitomicin reduced cell motility and potentiated the anti-motility activity of paclitaxel. The inhibitory effect was further potentiated by the addition of the HDAC6 inhibitor trichostatin A.\n",
      "\n",
      "\n",
      "Cell motility and adhesion involves dynamic microtubule (MT) acetylation/deacetylation, a process regulated by enzymes as HDAC6, a major cytoplasmic α-tubulin deacetylase.\n",
      "\n",
      "\n",
      "GRK2 and HDAC6 colocalize in the lamellipodia of migrating cells, leading to local tubulin deacetylation and enhanced motility.\n",
      "\n",
      "\n",
      "This review highlights the emerging roles of tubulin acetylation in multiple cellular functions, ranging from cell motility, cell cycle progression or cell differentiation to intracellular trafficking and signalling.\n",
      "\n",
      "\n",
      "Our results indicate that TPPP/p25 binds to HDAC6 (histone deacetylase 6), an enzyme responsible for tubulin deacetylation. Moreover, we demonstrated that the direct interaction of these two proteins resulted in the inhibition of the deacetylase activity of HDAC6.\n",
      "\n",
      "\n",
      "Finally, we demonstrated that, similarly to other HDAC6 inhibitors, TPPP/p25 influences the microtubule dynamics by decreasing the growth velocity of the microtubule plus ends and also affects cell motility as demonstrated by time lapse video experiments.\n",
      "\n",
      "\n",
      "\"tubacin,\" which inhibits alpha-tubulin deacetylation in mammalian cells.\n",
      "\n",
      "\n",
      "We provide evidence that class II histone deacetylase 6 (HDAC6) is the intracellular target of tubacin.\n",
      "\n",
      "\n",
      "Tubacin treatment did not affect the stability of microtubules but did decrease cell motility.\n",
      "\n",
      "\n",
      "They also suggest that small molecules that selectively inhibit HDAC6-mediated alpha-tubulin deacetylation, a first example of which is tubacin, might have therapeutic applications as antimetastatic and antiangiogenic agents.\n",
      "\n",
      "\n",
      "Furthermore, overexpression of HDAC6 promotes chemotactic cell movement, supporting the idea that HDAC6-mediated deacetylation regulates microtubule-dependent cell motility.\n",
      "\n",
      "\n",
      "HDAC6 is a major cytoplasmic a-tubulin deacetylase that is involved in cell motility and adhesion. GRK2 dynamically and directly associates with and phosphorylates HDAC6 to stimulate its a-tubulin deacetylase activity at specific cellular localizations, such as the leading edge of migrating cells, thus promoting local tubulin deacetylation and enhanced motility.\n",
      "\n",
      "\n",
      "====================================================================================================\n",
      "\n",
      "Result #4\n",
      "\n",
      "------\n",
      "['No, NADPH oxidase 5 is not expressed in rodents, because the gene is absent.']\n",
      "\n",
      "\n",
      "\n",
      "Because the Nox5 gene is absent in rodents, we generated transgenic mice expressing human Nox5 in a podocyte-specific manner (Nox5(pod+)). \n",
      "\n",
      "\n",
      "The NADPH oxidase 5 (NOX5) gene is present in humans but not rodents. \n",
      "\n",
      "\n",
      "The data document that the NOX5 gene was expressed in cells of lagomorphs unlike rodents, making the rabbit an interesting model to study NOX5 functions.\n",
      "\n",
      "\n",
      "Nox5 was lost in rodents, and Nox3, which functions in the inner ear in gravity perception, emerged the most recently, corresponding to full-time adaptation of vertebrates to land. \n",
      "\n",
      "\n",
      "NOX expression patterns in animals are complex and ancestral NOXes, NOX5-like isoforms and DUOXes are generally found. But there are exceptions; for example rodents lack NOX5 and Caenorhabditis elegans expresses only DUOXes.\n",
      "\n",
      "\n",
      "The NADPH oxidase 5 (NOX5) gene is present in humans but not rodents.\n",
      "\n",
      "\n",
      "The NADPH oxidase 5 (NOX5) gene is present in humans but not rodents\n",
      "\n",
      "\n",
      "NADPH oxidases are the major sources of reactive oxygen species in cardiovascular, neural, and kidney cells. The NADPH oxidase 5 (NOX5) gene is present in humans but not rodents.\n",
      "\n",
      "\n",
      "But there are exceptions; for example rodents lack NOX5 and Caenorhabditis elegans expresses only DUOXes.\n",
      "\n",
      "\n",
      "Because the Nox5 gene is absent in rodents, we generated transgenic mice expressing human Nox5 in a podocyte-specific manner (Nox5(pod+)).\n",
      "\n",
      "\n",
      "The data document that the NOX5 gene was expressed in cells of lagomorphs unlike rodents, making the rabbit an interesting model to study NOX5 functions.\n",
      "\n",
      "\n",
      "The most recently identified member of the Nox family, Nox5, has for the most part been overlooked in renal disease, partly owing to its absence from the rodent genome.\n",
      "\n",
      "\n",
      "====================================================================================================\n",
      "\n",
      "Result #5\n",
      "\n",
      "------\n",
      "['Phospholamban (PLB) is a sarcoplasmic reticulum (SR) protein that is phosphorylated at Ser16 by PKA. Phosphorylation of PLB by PKA reverses the inhibitory action of PLB.']\n",
      "\n",
      "\n",
      "\n",
      "cAMP-dependent protein kinase (PKA) phosphorylation of PLB\n",
      "\n",
      "\n",
      "phosphorylation of PLN, at either Ser(16) by PKA \n",
      "\n",
      "\n",
      "Activation of cardiac muscle sarcoplasmic reticulum Ca2+-ATPase (SERCA2a) by beta1-agonists involves cAMP- and PKA-dependent phosphorylation of phospholamban (PLB), which relieves the inhibitory effects of PLB on SERCA2a. \n",
      "\n",
      "\n",
      "Phospholamban (PLB) is a sarcoplasmic reticulum (SR) protein that when phosphorylated at Ser16 by PKA\n",
      "\n",
      "\n",
      "phosphorylation of PLB by the Ca2+-calmodulin-dependent protein kinase (CaMK) and cAMP-dependent protein kinase (PKA). \n",
      "\n",
      "\n",
      "cAMP-dependent protein kinase (PKA)-mediated phospholamban (PLB) phosphorylation at serine-16\n",
      "\n",
      "\n",
      "Phospholamban (PLB) is a major target of the beta-adrenergic cascade in the heart, functioning to modulate contractile force by altering the rate of calcium re-sequestration by the Ca-ATPase. Functionally, inhibition by PLB binding is manifested by shifts in the calcium dependence of Ca-ATPase activation toward higher calcium levels; phosphorylation of PLB by PKA reverses the inhibitory action of PLB.\n",
      "\n",
      "\n",
      "phosphorylation of both PLB residues (Ser16, PKA site, and Thr17, CaMKII site) \n",
      "\n",
      "\n",
      "Phosphorylation of Ser(16) by PKA\n",
      "\n",
      "\n",
      "stabilization of the structure of PLB following phosphorylation of Ser(16)\n",
      "\n",
      "\n",
      "Phospholamban (PLB) inhibits the sarcoplasmic reticulum (SR) Ca(2+)-ATPase, and this inhibition is relieved by cAMP-dependent protein kinase (PKA)-mediated phosphorylation. \n",
      "\n",
      "\n",
      "Two-dimensional tryptic peptide maps of phosphorylated phospholamban indicated that cAMP-dependent protein kinase phosphorylates at a single site, A, and Ca2+-calmodulin-dependent protein kinase phosphorylates at sites C1 and C2 in the low molecular weight form, where A is different from C1 but may be the same as C2.\n",
      "\n",
      "\n",
      "Because SR function is regulated by phosphorylation of phospholamban (PLB), a SR protein phosphorylated by cAMP-dependent protein kinase (PKA) at Ser(16)and Ca(2+)-calmodulin-dependent protein kinase (CaMKII) at Thr(17), the phosphorylation of these residues during ischemia and reperfusion was examined in Langendorff-perfused rat hearts\n",
      "\n",
      "\n",
      "These changes were associated with reduced protein expression of sarcoplasmic reticulum Ca(2+)-ATPase (SERCA2a) and protein kinase A phosphorylated phospholamban (PLB), which was reduced in HF, but essentially abolished in VD-HF\n",
      "\n",
      "\n",
      "The data indicate that 1) phosphorylation of phospholamban at Ser16 by cAMP-dependent protein kinase is the main regulator of beta-adrenergic-induced cardiac relaxation definitely preceding Thr17 phosphorylation and 2) the beta-adrenergic-mediated phosphorylation of Thr17 by Ca2+-calmodulin-dependent protein kinase required influx of Ca2+ through the L-type Ca2+ channel\n",
      "\n",
      "\n",
      "Here we extend this model to explain the reversal of SERCA2a inhibition that occurs after phosphorylation of PLB at Ser(16) by protein kinase A (PKA) and after binding of the anti-PLB monoclonal antibody 2D12, which recognizes residues 7-13 of PLB\n",
      "\n",
      "\n",
      "Phospholamban is phosphorylated in heart by cAMP-dependent protein kinase, cGMP-dependent protein kinase and calcium/calmodulin-dependent protein kinase II (CM-kinase-II) and in smooth muscle cells by cGMP-dependent protein kinase\n",
      "\n",
      "\n",
      "Phospholamban, the cardiac sarcoplasmic reticulum proteolipid, is phosphorylated by cAMP-dependent protein kinase, by Ca2+/phospholipid-dependent protein kinase, and by an endogenous Ca2+/calmodulin-dependent protein kinase, the identity of which remains to be defined\n",
      "\n",
      "\n",
      "====================================================================================================\n",
      "\n",
      "Result #6\n",
      "\n",
      "------\n",
      "['Yes, vitamin D induces autophagy.']\n",
      "\n",
      "\n",
      "\n",
      " 1,25(OH)2D treatment was accompanied by autophagy activation \n",
      "\n",
      "\n",
      "Autophagy signaling pathway was regulated by vitamin D3\n",
      "\n",
      "\n",
      "vitamin D induces autophagy\n",
      "\n",
      "\n",
      "Vitamin D shows promise for the prevention and amelioration of pathologic responses in IBD, an effect that is mediated, at least in part, by the induction and modulation of autophagy.\n",
      "\n",
      "\n",
      "====================================================================================================\n",
      "\n",
      "Result #7\n",
      "\n",
      "------\n",
      "['Yes, junctin binds to ryanodine receptors within the junctional sarcoplasmic reticulum of calcium release units, and normally acts as an activator of RyR channels at low luminal [Ca(2+)], and as an inhibitor at high luminal [Ca(2+)].']\n",
      "\n",
      "\n",
      "\n",
      "Junctin, a 26 kDa intra-sarcoplasmic reticulum (SR) protein, forms a quaternary complex with triadin, calsequestrin and the ryanodine receptor (RyR) at the junctional SR membrane. \n",
      "\n",
      "\n",
      "Junctin ablation appears to affect how RyRs 'sense' SR Ca(2+) load, resulting in decreased diastolic SR Ca(2+) leak despite an elevated [Ca(2+)](SR). \n",
      "\n",
      "\n",
      "Single channel recordings of RyRs from WT and JCN-KO cardiac SR indicate that the absence of junctin produces a dual effect on the normally linear response of RyRs to luminal [Ca(2+)]: at low luminal [Ca(2+)] (<1 mmol l(-1)), junctin-devoid RyR channels are less responsive to luminal [Ca(2+)]; conversely, high luminal [Ca(2+)] turns them hypersensitive to this form of channel modulation. Thus, junctin produces complex effects on Ca(2+) sparks, transients, and leak, but the luminal [Ca(2+)]-dependent dual response of junctin-devoid RyRs demonstrates that junctin normally acts as an activator of RyR channels at low luminal [Ca(2+)], and as an inhibitor at high luminal [Ca(2+)].\n",
      "\n",
      "\n",
      "Normal Ca(2+) signalling in skeletal muscle depends on the membrane associated proteins triadin and junctin and their ability to mediate functional interactions between the Ca(2+) binding protein calsequestrin and the type 1 ryanodine receptor in the lumen of the sarcoplasmic reticulum.\n",
      "\n",
      "\n",
      "We show here that purified skeletal ryanodine receptors are similarly activated by purified triadin or purified junctin added to their luminal side, although a lack of competition indicated that the proteins act at independent sites. Surprisingly, triadin and junctin differed markedly in their ability to transmit information between skeletal calsequestrin and ryanodine receptors. Purified calsequestrin inhibited junctin/triadin-associated, or junctin-associated, ryanodine receptors and the calsequestrin re-associated channel complexes were further inhibited when luminal Ca(2+) fell from 1mM to\n",
      "\n",
      "\n",
      "By fusing GCaMP6f to the N-terminus of triadin 1 or junctin, GCaMP6f-T/J was targeted to dyadic junctions, where it colocalized with t-tubules and RyRs after adenovirus-mediated gene transfer. \n",
      "\n",
      "\n",
      "The junctional face of the jSR, facing the transverse tubules, is occupied by a molecular complex composed of the transmembrane Ca2+ release channels (ryanodine receptors); the luminal protein calsequestrin (CSQ); the 2 membrane proteins, junctin (Jct), and triadin (Tr), which mediate CSQ-ryanodine receptor interactions; and several other components.\n",
      "\n",
      "\n",
      "Calsequestrin, the main calcium buffer in the sarcoplasmic reticulum, provides a pool of calcium for release through the ryanodine receptor and acts as a luminal calcium sensor for the channel via its interactions with triadin and junctin. We examined the influence of phosphorylation of calsequestrin on its ability to store calcium, to polymerise and to regulate ryanodine receptors by binding to triadin and junctin. \n",
      "\n",
      "\n",
      "Junctin is a 26 kDa membrane protein that binds to calsequestrin, triadin, and ryanodine receptors (RyRs) within the junctional sarcoplasmic reticulum of calcium release units. \n",
      "\n",
      "\n",
      "====================================================================================================\n",
      "\n",
      "Result #8\n",
      "\n",
      "------\n",
      "['Yes, fbroblast growth factor 23 (FGF23) is a phosphaturic hormone.']\n",
      "\n",
      "\n",
      "\n",
      "PTH can induce skeletal synthesis of another potent phosphaturic hormone, fibroblast growth factor 23 (FGF23), \n",
      "\n",
      "\n",
      " Fibroblast growth factor 23 (FGF23) is a phosphaturic hormone that has recently been identified as a CKD-related factor affecting CRS. \n",
      "\n",
      "\n",
      " circulating phosphaturic hormone fibroblast growth factor-23 levels\n",
      "\n",
      "\n",
      " Fibroblast growth factor (FGF) 23 is one of the most recently discovered FGFs. This phosphaturic hormone produced in bones is a risk factor for cardiovascular diseases and thus mortality.\n",
      "\n",
      "\n",
      "fibroblast growth factor 23 (FGF23), a phosphaturic hormone and regulator of 1,25(OH)2 vitamin D3 (1,25VitD3). \n",
      "\n",
      "\n",
      "fibroblast growth factor-23 (FGF23), a bone-derived phosphaturic hormone.\n",
      "\n",
      "\n",
      " the phosphaturic hormone fibroblast growth factor 23 (FGF23) and soluble Klotho with all-cause mortality.\n",
      "\n",
      "\n",
      "In particular, diseases caused by changes in the expression and proteolytic control of the phosphaturic hormone fibroblast growth factor-23 (FGF23) have come to the forefront in terms of directing new models explaining mineral metabolism\n",
      "\n",
      "\n",
      "serum levels of a phosphaturic hormone, fibroblast growth factor 23 (Fgf23), \n",
      "\n",
      "\n",
      "====================================================================================================\n",
      "\n",
      "Result #9\n",
      "\n",
      "------\n",
      "['Here, we present our results showing a connection between the linker histones, the higher-order chromatin structures, and the process of chronological lifespan of yeast cells. Characteristically, linker histone depleted chromatin generally exhibited longer chromatin loops than the wild-type. These results suggest that HHO1p may play a similar role to linker histones, but at restricted locations in the chromatin. The binding was structure specific, since the use of double-stranded DNA, or a mutant Hho1p in which the second DNA binding site of globular domain 1 was abolished, resulted in a significant decrease in bridged binding.', 'Hho1p is a bona fide linker histone', 'In Saccharomyces cerevisiae, HHO1 encodes a putative linker histone with very significant homology to histone H1. The putative linker histone in Saccharomyces cerevisiae, Hho1p, has two regions of sequence (GI and GII) that are homologous to the single globular domains of linker histones H1 and H5 in higher eukaryotes.']\n",
      "\n",
      "\n",
      "\n",
      "Hho1p is a bona fide linker histone\n",
      "\n",
      "\n",
      "In Saccharomyces cerevisiae, HHO1 encodes a putative linker histone with very significant homology to histone H1\n",
      "\n",
      "\n",
      "HHO1p may play a similar role to linker histones, but at restricted locations in the chromatin\n",
      "\n",
      "\n",
      "The putative linker histone in Saccharomyces cerevisiae, Hho1p, has two regions of sequence (GI and GII) that are homologous to the single globular domains of linker histones H1 and H5 in higher eukaryotes. \n",
      "\n",
      "\n",
      "The Saccharomyces cerevisiae homologue of the linker histone H1, Hho1p, has two domains that are similar in sequence to the globular domain of H1 (and variants such as H5)\n",
      "\n",
      "\n",
      "Two homologous domains of similar structure but different stability in the yeast linker histone, Hho1p\n",
      "\n",
      "\n",
      "Saccharomyces cerevisiae encodes a single linker histone, Hho1p, with two globular domains. \n",
      "\n",
      "\n",
      "The Saccharomyces cerevisiae linker histone Hho1p, with two globular domains, can simultaneously bind to two four-way junction DNA molecules\n",
      "\n",
      "\n",
      "Here, we show in yeast, that the presence of yeast linker histone Hho1p represses expression of a pol II transcribed gene (MET15) embedded in the rDNA.\n",
      "\n",
      "\n",
      "Yeast linker histone Hho1p is required for efficient RNA polymerase I processivity and transcriptional silencing at the ribosomal DNA\n",
      "\n",
      "\n",
      "Saccharomyces cerevisiae linker histone Hho1p is not essential for cell viability, and very little is known about its function in vivo. \n",
      "\n",
      "\n",
      "Saccharomyces cerevisiae linker histone Hho1p functionally interacts with core histone H4 and negatively regulates the establishment of transcriptionally silent chromatin\n",
      "\n",
      "\n",
      " Unlike canonical linker histones in higher eukaryotes that have a single conserved globular domain, Hho1p possesses two globular domains. We show that the carboxyl-terminal globular domain of Hho1p is dispensable for its function, suggesting that the mode of Hho1p action is similar to that of canonical linker histones\n",
      "\n",
      "\n",
      "To identify new proteins involved in spore nuclear organization, we purified chromatin from mature spores and discovered a significant enrichment of the linker histone (Hho1)\n",
      "\n",
      "\n",
      "Hho1 chromatin immunoprecipitation followed by sequencing (ChIP-seq) revealed increased genome-wide binding in mature spores and provides novel in vivo evidence of the linker histone binding to nucleosomal linker DNA\n",
      "\n",
      "\n",
      "One of the peculiarities of S. cerevisiae cells is the unusual and less abundant linker histone, Hho1p.\n",
      "\n",
      "\n",
      "Hho1p, the linker histone of Saccharomyces cerevisiae, is important for the proper chromatin organization in vivo\n",
      "\n",
      "\n",
      "Characteristically, linker histone depleted chromatin generally exhibited longer chromatin loops than the wild-type. \n",
      "\n",
      "\n",
      "Saccharomyces cerevisiae linker histone-Hho1p maintains chromatin loop organization during ageing.\n",
      "\n",
      "\n",
      "Database homology searching against the complete yeast genome has identified a gene, HHO1, (or YPL127C, formerly LPI17) which encodes a protein that has two regions that show similarity to the pea histone H1 globular domain.\n",
      "\n",
      "\n",
      "Database homology searching against the complete yeast genome has identified a gene, HHO1, (or YPL127C, formerly LPI17) which encodes a protein that has two regions that show similarity to the pea histone H1 globular domain. \n",
      "\n",
      "\n",
      "Biochemical studies to date have not been able to identify the linker histone H1 protein in the budding yeast Saccharomyces cerevisiae. Database homology searching against the complete yeast genome has identified a gene, HHO1, (or YPL127C, formerly LPI17) which encodes a protein that has two regions that show similarity to the pea histone H1 globular domain.\n",
      "\n",
      "\n",
      "Database homology searching against the complete yeast genome has identified a gene, HHO1, (or YPL127C, formerly LPI17) which encodes a protein that has two regions that show similarity to the pea histone H1 globular domain.\n",
      "\n",
      "\n",
      "====================================================================================================\n",
      "\n",
      "Result #10\n",
      "\n",
      "------\n",
      "['Yes, hypersensitivity to DNA crosslinking agents is one of the hallmarks of Fanconi anemia, the others being defective FANCD2 monoubiquitination, and genomic instability.']\n",
      "\n",
      "\n",
      "\n",
      "The Fanconi anemia (FA) core complex plays a central role in the DNA damage response network\n",
      "\n",
      "\n",
      "FAAP100-deficient cells display hallmark features of FA cells, including defective FANCD2 monoubiquitination, hypersensitivity to DNA crosslinking agents, and genomic instability.\n",
      "\n",
      "\n",
      "Fanconi anemia (FA) is a rare genetic disorder characterized by aplastic anemia, cancer/leukemia susceptibility and cellular hypersensitivity to DNA crosslinking agents, such as cisplatin.\n",
      "\n",
      "\n",
      "Fanconi anemia (FA) is an inherited chromosomal recessive syndrome characterized by cellular hypersensitivity to DNA crosslinking agents and bone marrow failure, which cause aplastic anemia, and an increased incidence of malignancy.\n",
      "\n",
      "\n",
      "Features of chromosomal aberrations, hypersensitivity to DNA crosslinking agents, and predisposition to malignancy have suggested a fundamental anomaly of DNA repair in Fanconi anemia.\n",
      "\n",
      "\n",
      "Fanconi anemia (FA) is one of several genetic diseases with characteristic cellular hypersensitivity to DNA crosslinking agents which suggest that FA proteins may function as part of DNA repair processes.\n",
      "\n",
      "\n",
      "Fanconi anemia (FA) is characterized by cellular hypersensitivity to DNA crosslinking agents, but how the Fanconi pathway protects cells from DNA crosslinks and whether FA proteins act directly on crosslinks remain unclear.\n",
      "\n",
      "\n",
      "Fanconi Anemia (FA) is a rare genetic disorder associated with a bone-marrow failure, cancer predisposition and hypersensitivity to DNA crosslinking agents.\n",
      "\n",
      "\n",
      "Fanconi anemia (FA) is a heterogeneous disease associated with a bone marrow failure, cancer predisposition and hypersensitivity to DNA crosslinking agents.\n",
      "\n",
      "\n",
      "Fanconi anemia (FA) is an inherited disorder characterized by defective DNA repair and cellular sensitivity to DNA crosslinking agents.\n",
      "\n",
      "\n",
      "Fanconi anemia (FA) is an inherited disease characterized by bone marrow failure, increased cancer risk and hypersensitivity to DNA cross-linking agents, implying a role for this pathway in the maintenance of genomic stability.\n",
      "\n",
      "\n",
      "Genetic or epigenetic inactivation of the pathway formed by the Fanconi Anemia (FA) proteins occurs in several cancer types, including head and neck squamous cell carcinomas (HNSCC), rendering the affected tumors potentially hypersensitive to DNA crosslinking agents.\n",
      "\n",
      "\n",
      "Fanconi Anemia (FA) is a rare autosomic recessive and X-linked disease with chromosomal instability after exposure to crosslinking agents as the hallmark.\n",
      "\n",
      "\n",
      "Fanconi Anemia (FA) is an autosomal recessive disease characterized by chromosome instability, cellular hypersensitivity to DNA cross-linking agents, and increased predisposition to malignancies.\n",
      "\n",
      "\n",
      "The Bloom protein (BLM) and Topoisomerase IIIalpha are found in association with proteins of the Fanconi anemia (FA) pathway, a disorder manifesting increased cellular sensitivity to DNA crosslinking agents.\n",
      "\n",
      "\n",
      "Using siRNA depletion and gene knockout techniques, we show that FAAP100-deficient cells display hallmark features of FA cells, including defective FANCD2 monoubiquitination, hypersensitivity to DNA crosslinking agents, and genomic instability.\n",
      "\n",
      "\n",
      "Fanconi anemia (FA) is a recessive human cancer prone syndrome featuring bone marrow failure, developmental abnormalities and hypersensitivity to DNA crosslinking agents exposure.\n",
      "\n",
      "\n",
      "FA is a chromosome instability syndrome characterized by childhood-onset aplastic anemia, cancer or leukemia susceptibility, and cellular hypersensitivity to DNA crosslinking agents.\n",
      "\n",
      "\n",
      "Functional defects in the Fanconi pathway can result in a marked hypersensitivity to interstrand crosslinking agents, such as mitomycin C.\n",
      "\n",
      "\n",
      "At the cellular level, hypersensitivity to DNA interstrand crosslinks is the defining feature in Fanconi anemia.\n",
      "\n",
      "\n",
      "DNA crosslinking agents may led to DNA cross-linking lesion, and Fanconi anemia pathway plays a key role in repairing its cross-linking.\n",
      "\n",
      "\n",
      "Fanconi anemia (FA), an autosomal recessive disorder of children, is characterized by congenital or childhood aplastic anemia, multiple developmental anomalies, increased incidence of myeloid leukemia, increased spontaneous chromosome breakage, and cellular and chromosomal hypersensitivity to DNA bifunctional crosslinking and alkylating agents.\n",
      "\n",
      "\n",
      "elegans provides an excellent model system for the study of the Fanconi Anemia (FA), one of the hallmarks of which is sensitivity to interstrand crosslinking agents\n",
      "\n",
      "\n",
      "Using siRNA depletion and gene knockout techniques, we show that FAAP100-deficient cells display hallmark features of FA cells, including defective FANCD2 monoubiquitination, hypersensitivity to DNA crosslinking agents, and genomic instability\n",
      "\n",
      "\n",
      "One of the hallmark phenotypes of FA is cellular hypersensitivity to agents that induce DNA interstrand crosslinks (ICLs), such as mitomycin C (MMC)\n",
      "\n",
      "\n",
      "Furthermore, the cytological hallmark of FA, the DNA crosslink-induced radial chromosome formation, exemplifies an innate impairment in the repair of these particularly cytotoxic DNA lesions [A.D\n",
      "\n",
      "\n",
      "Fanconi anemia (FA) is characterized by cellular hypersensitivity to DNA crosslinking agents, but how the Fanconi pathway protects cells from DNA crosslinks and whether FA proteins act directly on crosslinks remain unclear\n",
      "\n",
      "\n",
      "Features of chromosomal aberrations, hypersensitivity to DNA crosslinking agents, and predisposition to malignancy have suggested a fundamental anomaly of DNA repair in Fanconi anemia\n",
      "\n",
      "\n",
      "Fanconi anemia (FA) is an inherited chromosomal recessive syndrome characterized by cellular hypersensitivity to DNA crosslinking agents and bone marrow failure, which cause aplastic anemia, and an increased incidence of malignancy\n",
      "\n",
      "\n",
      "Genetic or epigenetic inactivation of the pathway formed by the Fanconi Anemia (FA) proteins occurs in several cancer types, including head and neck squamous cell carcinomas (HNSCC), rendering the affected tumors potentially hypersensitive to DNA crosslinking agents\n",
      "\n",
      "\n",
      "Fanconi anemia (FA) is a human autosomal disorder characterized by cancer susceptibility and cellular sensitivity to DNA crosslinking agents such as mitomycin C and diepoxybutane\n",
      "\n",
      "\n",
      "The Fanconi anemia pathway promotes DNA glycosylase-dependent excision of interstrand DNA crosslinks.\n",
      "\n",
      "\n",
      "DNA crosslinking agents may led to DNA cross-linking lesion, and Fanconi anemia pathway plays a key role in repairing its cross-linking\n",
      "\n",
      "\n",
      "FA is a chromosome instability syndrome characterized by childhood-onset aplastic anemia, cancer or leukemia susceptibility, and cellular hypersensitivity to DNA crosslinking agents\n",
      "\n",
      "\n",
      "The disease is manifested by defects in DNA repair, hypersensitivity to DNA crosslinking agents, and a high degree of chromosomal aberrations\n",
      "\n",
      "\n",
      "====================================================================================================\n"
     ]
    }
   ],
   "source": [
    "executeQuery(\"Is RANKL secreted from the cells?\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 61,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Result #1\n",
      "\n",
      "------\n",
      "['There are not reported data indicating that metformin interferes with thyroxine absorption']\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "Result #2\n",
      "\n",
      "------\n",
      "['Myasthenia gravis (MG) is a neuromuscular disease which has been associated with an increased risk of glucocorticoid-induced osteoporosis. Thymectomy can also increase risk for osteoporosis. Appropriate osteoporosis preventive measures can reduce osteoporosis risk in MG patients.']\n",
      "\n",
      "\n",
      "\n",
      "We performed PVP in 4 patients with generalized MG associated with recent steroid-induced symptomatic VFs. \n",
      "\n",
      "\n",
      "In this case report, we used tacrolimus to successfully treat a 13-year-old boy with ocular MG who had suffered from severe steroid complications, including a failure of thrive and osteoporosis.\n",
      "\n",
      "\n",
      " INTRODUCTION: Myasthenia gravis (MG) is a neuromuscular disease which has been associated with an increased falls risk and glucocorticoid-induced osteoporosis, recognized determinants of increased fracture risk. \n",
      "\n",
      "\n",
      "RESULTS: Compared to the control cohort, there was no statistically significant increased risk observed in patients with MG for any fracture (adjusted hazard ratio [AHR] 1.11; 95 % confidence interval [CI], 0.84-1.47) or osteoporotic fractures (AHR 0.98 [95 % CI 0.67-1.41]). Further, use of oral glucocorticoids up to a cumulative dose exceeding 5 g prednisolone equivalents did not increase risk of osteoporotic fracture (AHR 0.99 [95 % CI, 0.31-3.14]) compared with MG patients without glucocorticoid exposure.\n",
      "\n",
      "\n",
      "The RANKL/OPG ratio and indices of bone metabolisms are also not affected by THX, although THX increases the levels of IL-7 and RANKL.\n",
      "\n",
      "\n",
      "Both disorders had been controlled for around 15 years by oral prednisolone and a cholinesterase inhibitor following surgical removal of invasive thymoma and radiotherapy, but muscular weakness due to myalgia and an increase in serum levels of myogenic enzymes, mainly ascribable to the recurrence of PM, reappeared immediately after cessation of these drugs, which was done because the patient had multiple bone fractures and severe osteoporosis due to the long-term corticosteroid therapy. \n",
      "\n",
      "\n",
      "We measured bone density in 36 patients (26 females and 10 males) who had undergone long-term prednisolone administration, and found a decrease in bone density in 31% of female patients and osteoporosis in only 11.5% (three cases).\n",
      "\n",
      "\n",
      "In conclusion, prednisolone-treated patients with myasthenia gravis have an acceptable risk of bone loss if prophylactic medication is administered.\n",
      "\n",
      "\n",
      "INTRODUCTION: Myasthenia gravis (MG) is a neuromuscular disease which has been associated with an increased falls risk and glucocorticoid-induced osteoporosis, recognized determinants of increased fracture risk.\n",
      "\n",
      "\n",
      "Alendronate should be used with caution in patients with myasthenia gravis who have corticosteroid-induced osteoporosis\n",
      "\n",
      "\n",
      "In this paper we present two cases of young women who developed severe PAO with vertebral fractures: a 42-year-old woman with a family history of osteoporosis, and a 21-year-old woman affected with myasthenia gravis\n",
      "\n",
      "\n",
      "Myasthenia gravis (MG) is a neuromuscular disease which has been associated with an increased falls risk and glucocorticoid-induced osteoporosis, recognized determinants of increased fracture risk\n",
      "\n",
      "\n",
      "====================================================================================================\n",
      "\n",
      "Result #3\n",
      "\n",
      "------\n",
      "['The detailed view of conservation across angiosperms revealed not only high coding-sequence conservation but also a large set of previously uncharacterized intergenic conservation. Grass genes have dramatically fewer and much smaller CNSs than mammalian genes. Using an alignment-free information-retrieval approach, we have comprehensively identified all long identical multispecies elements (LIMEs), which include both syntenic and nonsyntenic regions, of at least 100 identical base pairs shared by at least two genomes. Using a comparative genomics approach with four dicotyledonous plant species (Arabidopsis thaliana, papaya [Carica papaya], poplar [Populus trichocarpa], and grape [Vitis vinifera]), we detected hundreds of CNSs upstream of Arabidopsis genes. We consequently compared the genomes of Arabidopsis thaliana and rice, which diverged about 200 million years ago, and identified 25 ultraconserved elements that are longer than 100 bp. Using a local sequence alignment set to deliver only significant alignments, we found one or more CNSs in the noncoding regions of the majority of genes studied. ', 'Grass genes have dramatically fewer and much smaller CNSs than mammalian genes. The detailed view of conservation across angiosperms revealed not only high coding-sequence conservation but also a large set of previously uncharacterized intergenic conservation. Using an alignment-free information-retrieval approach, we have comprehensively identified all long identical multispecies elements (LIMEs), which include both syntenic and nonsyntenic regions, of at least 100 identical base pairs shared by at least two genomes. Using a comparative genomics approach with four dicotyledonous plant species (Arabidopsis thaliana, papaya [Carica papaya], poplar [Populus trichocarpa], and grape [Vitis vinifera]), we detected hundreds of CNSs upstream of Arabidopsis genes. We consequently compared the genomes of Arabidopsis thaliana and rice, which diverged about 200 million years ago, and identified 25 ultraconserved elements that are longer than 100 bp. Using a local sequence alignment set to deliver only significant alignments, we found one or more CNSs in the noncoding regions of the majority of genes studied. ', 'Yes. Conserved, UltraConserved and other classes of Constrained Noncoding Sequences have been found in plant genomes.']\n",
      "\n",
      "\n",
      "\n",
      "Conservation and functional element discovery in 20 angiosperm plant genomes\n",
      "\n",
      "\n",
      "The detailed view of conservation across angiosperms revealed not only high coding-sequence conservation but also a large set of previously uncharacterized intergenic conservation\n",
      "\n",
      "\n",
      "Conserved noncoding sequences highlight shared components of regulatory networks in dicotyledonous plants\n",
      "\n",
      "\n",
      "Using a comparative genomics approach with four dicotyledonous plant species (Arabidopsis thaliana, papaya [Carica papaya], poplar [Populus trichocarpa], and grape [Vitis vinifera]), we detected hundreds of CNSs upstream of Arabidopsis genes\n",
      "\n",
      "\n",
      "Long identical multispecies elements in plant and animal genomes.\n",
      "\n",
      "\n",
      "Using an alignment-free information-retrieval approach, we have comprehensively identified all long identical multispecies elements (LIMEs), which include both syntenic and nonsyntenic regions, of at least 100 identical base pairs shared by at least two genomes\n",
      "\n",
      "\n",
      "In contrast, among six plant genomes, we only found nonsyntenic LIMEs\n",
      "\n",
      "\n",
      " Although complex LIMEs were found in both animal and plant genomes, they differed significantly in their composition and copy number\n",
      "\n",
      "\n",
      "Ultraconserved elements between the genomes of the plants Arabidopsis thaliana and rice\n",
      "\n",
      "\n",
      "We consequently compared the genomes of Arabidopsis thaliana and rice, which diverged about 200 million years ago, and identified 25 ultraconserved elements that are longer than 100 bp\n",
      "\n",
      "\n",
      "ultraconserved elements in plants tend to occur in clusters and locate at noncoding regions\n",
      "\n",
      "\n",
      "the functions of these plant ultraconserved elements and the reasons why they are practically frozen during the evolution of millions of years remain a mystery\n",
      "\n",
      "\n",
      "Conserved noncoding sequences in the grasses\n",
      "\n",
      "\n",
      " Using a local sequence alignment set to deliver only significant alignments, we found one or more CNSs in the noncoding regions of the majority of genes studied. Grass genes have dramatically fewer and much smaller CNSs than mammalian genes\n",
      "\n",
      "\n",
      "Conserved noncoding sequences among cultivated cereal genomes identify candidate regulatory sequence elements and patterns of promoter evolution\n",
      "\n",
      "\n",
      "Surveys for conserved noncoding sequences (CNS) among genes from monocot cereal species were conducted to assess the general properties of CNS in grass genomes and their correlation with known promoter regulatory elements\n",
      "\n",
      "\n",
      "Comparisons of orthologous maize-rice and maize-sorghum gene pairs identified 20 bp as a minimal length criterion for a significant CNS among grass genes, with few such CNS found to be conserved across rice, maize, sorghum, and barley\n",
      "\n",
      "\n",
      "====================================================================================================\n",
      "\n",
      "Result #4\n",
      "\n",
      "------\n",
      "['Yes, venlafaxine inhibits both the NET and SERT.']\n",
      "\n",
      "\n",
      "\n",
      "Treatment for 14 days with 70 mg/kg per day venlafaxine, which inhibits both the NET and SERT, or 10 mg/kg per day phenelzine, a monoamine oxidase inhibitor, produced antidepressant-like effects on behavior without altering NET or SERT expression.\n",
      "\n",
      "\n",
      "Venlafaxine blocks both serotonin and norepinephrine transporters (SERT and NET), with higher affinity for SERT.\n",
      "\n",
      "\n",
      "Chronic venlafaxine treatment affected SERT and NET binding differently from paroxetine or desipramine.\n",
      "\n",
      "\n",
      "Venlafaxine blocks both serotonin and norepinephrine transporters (SERT and NET), with higher affinity for SERT\n",
      "\n",
      "\n",
      "Paroxetine and venlafaxine are potent serotonin transporter (SERT) antagonists and weaker norepinephrine transporter (NET) antagonists\n",
      "\n",
      "\n",
      "Using a novel blood assay that estimates CNS transporter occupancy we estimated the relative SERT and NET occupancy of paroxetine and venlafaxine in human subjects to assess the relative magnitude of SERT and NET inhibition\n",
      "\n",
      "\n",
      "Treatment for 14 days with 70 mg/kg per day venlafaxine, which inhibits both the NET and SERT, or 10 mg/kg per day phenelzine, a monoamine oxidase inhibitor, produced antidepressant-like effects on behavior without altering NET or SERT expression\n",
      "\n",
      "\n",
      "We then performed the first reported investigation of epistasis between the SERT gene and norepinephrine transporter gene (SLC6A2, alias NET) in AN, as an earlier study suggested that atypical AN responds to the dual serotonin-norepinephrine reuptake inhibitor venlafaxine\n",
      "\n",
      "\n",
      "Of particular interest were the findings that paroxetine, generally thought of as a selective SERT antagonist, possesses moderately high affinity for the NET and that venlafaxine, which has been described as a &quot;dual uptake inhibitor&quot;, possesses weak affinity for the NET\n",
      "\n",
      "\n",
      "The ratios of measured occupancy ED(50) values (doses at which 50% occupancy occurs) among SERT, NET and DAT sites for duloxetine, venlafaxine, nomifensine, indatraline, DOV 21,947 and DOV 216,303 were consistent with the ratios of the in vitro affinities between these target binding sites\n",
      "\n",
      "\n",
      "SERT and NET occupancy by venlafaxine and milnacipran in nonhuman primates: a PET study\n",
      "\n",
      "\n",
      "In this study in nonhuman primates, we aimed to investigate the relationship between SERT and NET affinity by measuring the in vivo occupancy at both transporters of venlafaxine and milnacipran\n",
      "\n",
      "\n",
      "We hypothesized that venlafaxine would affect monoamine transporters dose-dependently, with low doses causing selective reduction of SERT binding sites and higher doses reducing both SERT and NET binding sites\n",
      "\n",
      "\n",
      "Comparative studies with clinically used antidepressants showed that venlafaxine possessed a profile similar to S33005 but was less potent. Clomipramine likewise interacted with SERTs and NETs but also with several other receptors types, while citalopram and reboxetine were preferential ligands of SERTs and NETs, respectively. In conclusion, S33005 interacts potently with SERTs and, less markedly, with NETs. \n",
      "\n",
      "\n",
      "Venlafaxine blocks both serotonin and norepinephrine transporters (SERT and NET), with higher affinity for SERT. Serotonergic effects occur with lower doses, whereas both serotonergic and noradrenergic effects occur with higher doses of venlafaxine.\n",
      "\n",
      "\n",
      "Taken together, the results from this study indicate that the low dose of venlafaxine blocked selectively the reuptake of 5-HT, whereas the high dose blocked the reuptake of both 5-HT and NE. Moreover, an enhancement of serotonergic neurotransmission by venlafaxine was only achieved under conditions whereby the desensitization of the terminal 5-HT(1B) autoreceptor is appended to that of the somatodendritic 5-HT(1A) receptor.\n",
      "\n",
      "\n",
      "====================================================================================================\n",
      "\n",
      "Result #5\n",
      "\n",
      "------\n",
      "['NO.']\n",
      "\n",
      "\n",
      "\n",
      "Amiodarone resulted in increased T4, T4/T3 and rT3, whereas dronedarone did not alter the thyroid hormone profile in normal animals.\n",
      "\n",
      "\n",
      "Fifty-five Wistar rats were randomly allocated to a 2-week oral treatment with either vehicle (n=18), amiodarone (30 mg/kg, n=20), or dronedarone (30 mg/kg, n=17).\n",
      "\n",
      "\n",
      "Thyroid function was similar in the 3 groups.\n",
      "\n",
      "\n",
      "Plasma levels of T3, T4, and rT3 were changed after SR 33589 treatment except a decrease in T4 level at the highest dose whilst the T4 T3 ratio and the level of rT3 were dose-dependently increased by amiodarone treatment.\n",
      "\n",
      "\n",
      "====================================================================================================\n",
      "\n",
      "Result #6\n",
      "\n",
      "------\n",
      "['Yes, RG7112 is a small molecule MDM2 antagonist.']\n",
      "\n",
      "\n",
      "\n",
      "To assess the influence of the p53 regulatory pathway further, we studied the effect of RG7112, a small molecule MDM2 antagonist that activates p53 by preventing its interaction with MDM2, on normal megakaryocytopoiesis and platelet production.\n",
      "\n",
      "\n",
      "RG7112 (2g) is the first clinical small-molecule MDM2 inhibitor designed to occupy the p53-binding pocket of MDM2.\n",
      "\n",
      "\n",
      "Effect of the MDM2 antagonist RG7112 on the P53 pathway in patients with MDM2-amplified, well-differentiated or dedifferentiated liposarcoma: an exploratory proof-of-mechanism study.\n",
      "\n",
      "\n",
      "We report a proof-of-mechanism study of RG7112, a small-molecule MDM2 antagonist, in patients with chemotherapy-naive primary or relapsed well-differentiated or dedifferentiated MDM2-amplified liposarcoma who were eligible for resection.\n",
      "\n",
      "\n",
      "To assess the influence of the p53 regulatory pathway further, we studied the effect of RG7112, a small molecule MDM2 antagonist that activates p53 by preventing its interaction with MDM2, on normal megakaryocytopoiesis and platelet production.\n",
      "\n",
      "\n",
      "Discovery of RG7112: A Small-Molecule MDM2 Inhibitor in Clinical Development.\n",
      "\n",
      "\n",
      "RG7112 was the first small-molecule p53-MDM2 inhibitor in clinical development.\n",
      "\n",
      "\n",
      "RG7112 binds MDM2 with high affinity (K(D) ~ 11 nmol/L), blocking its interactions with p53 in vitro.\n",
      "\n",
      "\n",
      "RG7112 is a selective inhibitor of p53-MDM2 binding that frees p53 from negative control, activating the p53 pathway in cancer cells leading to cell cycle arrest and apoptosis.\n",
      "\n",
      "\n",
      "The effects of RG7112 and Peg-IFNα 2a on MPN progenitor cells were dependent on blocking p53-MDM2 interactions and activating the p53 pathway, thereby increasing MPN CD34(+) cell apoptosis\n",
      "\n",
      "\n",
      "The orally bioavailable MDM2 antagonist RG7112 and pegylated interferon α 2a target JAK2V617F-positive progenitor and stem cells\n",
      "\n",
      "\n",
      "Initial testing of the MDM2 inhibitor RG7112 by the Pediatric Preclinical Testing Program\n",
      "\n",
      "\n",
      "In this issue of Blood, Lu et al describe the cooperation between an orally bioavailable mouse double minute 2 (MDM2) antagonist (RG7112) and the pegylated interferon α (Peg-IFNα 2a) to target JAK2V617F hematopoietic progenitors and stem cells\n",
      "\n",
      "\n",
      "MDM2 small-molecule antagonist RG7112 activates p53 signaling and regresses human tumors in preclinical cancer models.\n",
      "\n",
      "\n",
      "Activation of p53 by the MDM2 inhibitor RG7112 impairs thrombopoiesis.\n",
      "\n",
      "\n",
      "The orally bioavailable MDM2 antagonist RG7112 and pegylated interferon α 2a target JAK2V617F-positive progenitor and stem cells.\n",
      "\n",
      "\n",
      "Initial testing of the MDM2 inhibitor RG7112 by the Pediatric Preclinical Testing Program.\n",
      "\n",
      "\n",
      "The primary endpoint was to assess markers of RG7112-dependent MDM2 inhibition and P53 pathway activation (P53, P21, MDM2, Ki-67, macrophage inhibitory cytokine-1 [MIC-1], and apoptosis). \n",
      "\n",
      "\n",
      "RG7112 is a selective inhibitor of p53-MDM2 binding that frees p53 from negative control, activating the p53 pathway in cancer cells leading to cell cycle arrest and apoptosis. RG7112 was selected for evaluation by the Pediatric Preclinical Testing Program (PPTP) due to the relatively low incidence of p53 mutations in pediatric cancers compared with adult malignancies.\n",
      "\n",
      "\n",
      "However, the hydrophobic protein-protein interaction surface represents a significant challenge for the development of small-molecule inhibitors with desirable pharmacological profiles. RG7112 was the first small-molecule p53-MDM2 inhibitor in clinical development.\n",
      "\n",
      "\n",
      "Treatment with low doses of RG7112, an orally available small-molecule inhibitor of p53-MDM2, both alone and combined with pegylated interferon α 2a (Peg-IFNα 2a), significantly decreased MPN colony-forming unit-granulocyte macrophage and burst-forming unit-erythroid numbers and preferentially eliminated the total number of JAKV617F(+) MPN hematopoietic progenitor cells. The effects of RG7112 and Peg-IFNα 2a on MPN progenitor cells were dependent on blocking p53-MDM2 interactions and activating the p53 pathway, thereby increasing MPN CD34(+) cell apoptosis.\n",
      "\n",
      "\n",
      "RG7112 (2g) is the first clinical small-molecule MDM2 inhibitor designed to occupy the p53-binding pocket of MDM2. In cancer cells expressing wild-type p53, RG7112 stabilizes p53 and activates the p53 pathway, leading to cell cycle arrest, apoptosis, and inhibition or regression of human tumor xenografts.\n",
      "\n",
      "\n",
      "====================================================================================================\n",
      "\n",
      "Result #7\n",
      "\n",
      "------\n",
      "['Yes, molindone has a tendency to cause weight loss or limited weight gain.']\n",
      "\n",
      "\n",
      "\n",
      "Mean weight increased by 0.54 kg, and mean body mass index by 0.24 kg/m(2). \n",
      "\n",
      "\n",
      "A large-scale trial comparing a first-generation antipsychotic (molindone) with newer agents did not find significant differences in treatment response, although the newer antipsychotics were associated with more severe weight gain. \n",
      "\n",
      "\n",
      "No agent demonstrated superior efficacy, and all were associated with side effects, including weight gain. \n",
      "\n",
      "\n",
      "The three treatment arms did not significantly differ in symptom decrease or time to discontinuation. Akathisia was more common with molindone and elevated prolactin concentrations more common with risperidone. Although weight gain and metabolic adverse events had occurred more often with olanzapine and risperidone during the acute trial, no significant between-drug differences emerged in most of these parameters during maintenance treatment. \n",
      "\n",
      "\n",
      "Olanzapine and risperidone were associated with significantly greater weight gain. Olanzapine showed the greatest risk of weight gain and significant increases in fasting cholesterol, low density lipoprotein, insulin, and liver transaminase levels. Molindone led to more self-reports of akathisia. \n",
      "\n",
      "\n",
      "Molindone is no more or less likely than typical drugs to cause movement disorders, but it does cause significantly more weight loss (2RCTs n=60 RR 2.78, CI 1.10 to 6.99, NNH 5 CI 2 to 77). \n",
      "\n",
      "\n",
      "Molindone may be an effective antipsychotic but its adverse effect profile does not differ significantly from that of typical antipsychotics (apart from the event of weight loss). \n",
      "\n",
      "\n",
      "Convergent evidence suggests a hierarchy in the magnitude of BWG that may be induced by diverse agents, being very high for clozapine and olanzapine; high for quetiapine, zotepin, chlorpromazine, and thioridazine; moderate for risperidone and sertindole; and low for ziprazidone, amisulpiride, haloperidol, fluphenazine, pimozide, and molindone. \n",
      "\n",
      "\n",
      "Loxapine and molindone induce weight decreases, and these exceptions are difficult to explain.\n",
      "\n",
      "\n",
      "It is no more or less likely than typical drugs to cause movement disorders, but causes significantly more weight loss (RR 2.78, CI 1.10 to 6.99).\n",
      "\n",
      "\n",
      "Molindone may be an effective antipsychotic; however, its adverse effect profile does not differ significantly from that of typical antipsychotics, apart from the event of weight loss. \n",
      "\n",
      "\n",
      "Among conventional agents, mean weight change ranged from a reduction of 0.39 kg with molindone to an increase of 3.19 kg with thioridazine.\n",
      "\n",
      "\n",
      "Weight gain has been reported with nearly every antipsychotic drug on the market (molindone is an exception). \n",
      "\n",
      "\n",
      "Although almost all antipsychotics induce bodyweight gain, molindone and loxapine appear to induce bodyweight loss. \n",
      "\n",
      "\n",
      "Clozapine and low-potency phenothiazines are associated with the largest gains and molindone with weight loss, but the mechanism is not known. \n",
      "\n",
      "\n",
      "On average, molindone patients lost 5 pounds over the 6 weeks of treatment, whereas thioridazine patients gained 6 pounds. \n",
      "\n",
      "\n",
      "Clinically, molindone has a tendency to cause weight loss and may have less effect on seizure threshold than conventional antipsychotic agents\n",
      "\n",
      "\n",
      " Monthly weights and neuroleptic dosages during the first three months of psychiatric hospitalization were compared between matched groups of patients receiving molindone, a combination of molindone and other neuroleptics, or other neuroleptic drugs. We found no significant differences in weight gain among the three groups. \n",
      "\n",
      "\n",
      "The weight-reducing property of molindone, a recently introduced antipsychotic drug, was tested in 9 hospitalized chronic schizophrenic patients. There was an average weight loss of 7.6 kg after 3 months on molindone; most of the loss occurred during the first month.\n",
      "\n",
      "\n",
      "====================================================================================================\n",
      "\n",
      "Result #8\n",
      "\n",
      "------\n",
      "['No data indicate causal connection between gastro esophageal/laryngopharyngeal(LPR)  reflux disease and the occurrence of intraoral burning sensations']\n",
      "\n",
      "\n",
      "\n",
      "Our results suggest that there is no causal connection between LPR episodes and the occurrence of intraoral burning sensations in the examined patients.\n",
      "\n",
      "\n",
      "As reported below, although this symptom may well be diagnostically misleading, careful diagnosis based on clinical signs may distinguish patients with BMS from those with reflux disease, and successful management of burning mouth is often enables.\n",
      "\n",
      "\n",
      "====================================================================================================\n",
      "\n",
      "Result #9\n",
      "\n",
      "------\n",
      "['Yes, amoxicillin is used for treatment of malnutrition in children.']\n",
      "\n",
      "\n",
      "\n",
      " Another RCT did not show superiority of ceftriaxone over amoxicilllin for these same outcomes, but adressed SAM children with and without complications (p = 0.27). Another RCT showed no difference between amoxicillin and cotrimoxazole efficacies for pneumonia in underweight, but not SAM. Our meta-analysis of 12 pooled susceptibility-studies for all types of bacterial isolates, including 2767 stricly SAM children, favoured amoxicillin over cotrimoxazole for susceptibility medians: 42% (IQR 27-55%) vs 22% (IQR 17-23%) and population-weighted-means 52.9% (range 23-57%) vs 35.4% (range 6.7-42%).\n",
      "\n",
      "\n",
      " Susceptibility-studies favour amoxicillin over cotrimoxazole. \n",
      "\n",
      "\n",
      "Oral amoxicillin for 5 days was as effective as intramuscular ceftriaxone for 2 days (1 RCT). For uncomplicated SAM, amoxicillin showed no benefit over placebo (1 retrospective study). \n",
      "\n",
      "\n",
      "Children who took amoxicillin and de-worming had 95% (HR = 1.95, 95%-CI = 1.17, 3.23) and 74% (HR = 1.74, 95%-CI = 1.07, 2.83) more probability to recover from SAM as compared to those who didn't take them.\n",
      "\n",
      "\n",
      "METHODS: In this randomized, double-blind, placebo-controlled trial, we randomly assigned Malawian children, 6 to 59 months of age, with severe acute malnutrition to receive amoxicillin, cefdinir, or placebo for 7 days in addition to ready-to-use therapeutic food for the outpatient treatment of uncomplicated severe acute malnutrition. \n",
      "\n",
      "\n",
      "In the amoxicillin, cefdinir, and placebo groups, 88.7%, 90.9%, and 85.1% of the children recovered, respectively (relative risk of treatment failure with placebo vs. amoxicillin, 1.32; 95% confidence interval [CI], 1.04 to 1.68; relative risk with placebo vs. cefdinir, 1.64; 95% CI, 1.27 to 2.11). The mortality rates for the three groups were 4.8%, 4.1%, and 7.4%, respectively (relative risk of death with placebo vs. amoxicillin, 1.55; 95% CI, 1.07 to 2.24; relative risk with placebo vs. cefdinir, 1.80; 95% CI, 1.22 to 2.64).\n",
      "\n",
      "\n",
      "Evaluation of the routine use of amoxicillin as part of the home-based treatment of severe acute malnutrition.\n",
      "\n",
      "\n",
      "OBJECTIVE: To determine whether the inclusion of amoxicillin correlates with better recovery rates in the home-based treatment of severe acute malnutrition with ready-to-use therapeutic food.\n",
      "\n",
      "\n",
      "The standard protocol group received a 7-day course of amoxicillin at the onset of treatment.\n",
      "\n",
      "\n",
      "RESULTS: Four hundred and ninety-eight children were treated according to the standard protocol with amoxicillin, and 1955 were treated under the alternate protocol without antibiotics. \n",
      "\n",
      "\n",
      "The recovery rate for children who received amoxicillin was worse at 4 weeks (40%vs. 71%) but similar after up to 12 weeks of therapy (84%vs. 86%), compared to the children treated without antibiotics.\n",
      "\n",
      "\n",
      "CONCLUSIONS: This review of two therapeutic feeding programmes suggests that children with severe acute malnutrition who were treated without amoxicillin did not have an inferior rate of recovery.\n",
      "\n",
      "\n",
      "Treatment of severe malnutrition with 2-day intramuscular ceftriaxone vs 5-day amoxicillin.\n",
      "\n",
      "\n",
      "To determine whether the inclusion of amoxicillin correlates with better recovery rates in the home-based treatment of severe acute malnutrition with ready-to-use therapeutic food.\n",
      "\n",
      "\n",
      "Evaluation of the routine use of amoxicillin as part of the home-based treatment of severe acute malnutrition\n",
      "\n",
      "\n",
      "OBJECTIVE: To determine whether the inclusion of amoxicillin correlates with better recovery rates in the home-based treatment of severe acute malnutrition with ready-to-use therapeutic food. METHODS: This retrospective cohort study compared data from the treatment of two groups of children in Malawi aged 6-59 months with uncomplicated severe acute malnutrition. \n",
      "\n",
      "\n",
      "The recovery rate for children who received amoxicillin was worse at 4 weeks (40%vs. 71%) but similar after up to 12 weeks of therapy (84%vs. 86%), compared to the children treated without antibiotics. Regression modelling indicated that this difference at 4 weeks was most strongly associated with the receipt of amoxicillin. CONCLUSIONS: This review of two therapeutic feeding programmes suggests that children with severe acute malnutrition who were treated without amoxicillin did not have an inferior rate of recovery. \n",
      "\n",
      "\n",
      "CONCLUSIONS: This review of two therapeutic feeding programmes suggests that children with severe acute malnutrition who were treated without amoxicillin did not have an inferior rate of recovery. \n",
      "\n",
      "\n",
      "====================================================================================================\n",
      "\n",
      "Result #10\n",
      "\n",
      "------\n",
      "['We demonstrate that the presence of long inverted repeats (IR) flanking the mouse Sry gene leads to the formation of the Sry circular transcript in cultured cells. We have found that in an in vitro assay, the SRY protein binds to several sites of the Sry gene and especially to a (CA)25 sequence and to a (CAG)30 repeat.', 'The presence of long inverted repeats (IR) flanking the mouse Sry gene leads to the formation of the Sry circular transcript in cultured cells.', 'We demonstrate that the presence of long inverted repeats (IR) flanking the mouse Sry gene leads to the formation of the Sry circular transcript in cultured cells Circularization requires the presence of both IR.', 'We demonstrate that the presence of long inverted repeats (IR) flanking the mouse Sry gene leads to the formation of the Sry circular transcript in cultured cells We have found that in an in vitro assay, the SRY protein binds to several sites of the Sry gene and especially to a (CA)25 sequence and to a (CAG)30 repeat', 'We demonstrate that the presence of long inverted repeats (IR) flanking the mouse Sry gene leads to the formation of the Sry circular transcript in cultured cells']\n",
      "\n",
      "\n",
      "\n",
      "We demonstrate that the presence of long inverted repeats (IR) flanking the mouse Sry gene leads to the formation of the Sry circular transcript in cultured cells\n",
      "\n",
      "\n",
      "Circularization requires the presence of both IR. As few as 400 complementary nt are necessary for this process\n",
      "\n",
      "\n",
      "The presence of the IR does not significantly stimulate intermolecular annealing and trans-splicing in vivo\n",
      "\n",
      "\n",
      "We have found that in an in vitro assay, the SRY protein binds to several sites of the Sry gene and especially to a (CA)25 sequence and to a (CAG)30 repeat\n",
      "\n",
      "\n",
      "The Q-rich domain of the mouse sex determining gene, Sry, is encoded by an in-frame insertion of a repetitive sequence composed of mostly CAG repeats.\n",
      "\n",
      "\n",
      "Inverted repeat structure of the Sry locus in mice.\n",
      "\n",
      "\n",
      "We performed separate amplifications of DXZ4 repetitive satellite sequences on the X chromosome, and SRY gene - testis determined factor on the Y chromosome, using nested PCR\n",
      "\n",
      "\n",
      "Detailed analysis of the Sry genomic locus reveals a further difference in that the mouse Sry open reading frame lies within 2.8 kilobases of unique sequence at the center of a large inverted repeat. \n",
      "\n",
      "\n",
      "Detailed analysis of the Sry genomic locus reveals a further difference in that the mouse Sry open reading frame lies within 2.8 kilobases of unique sequence at the center of a large inverted repeat.\n",
      "\n",
      "\n",
      "The mouse genomic Sry locus is characterized by two arms of a large inverted repeat, flanking a unique region that, between an acceptor and a donor splice site, contains a single exon encoding the Sry protein.\n",
      "\n",
      "\n",
      "Recombination involving the repeat region may have led to an 11-kilobase deletion, precisely excising Sry in a line of XY female mice.\n",
      "\n",
      "\n",
      "Repetitive element analysis revealed numerous LINE-L1 elements at regions where conservation is lost among the Sry copies.\n",
      "\n",
      "\n",
      "Inverted repeat structure of the Sry locus in mice.\n",
      "\n",
      "\n",
      "====================================================================================================\n"
     ]
    }
   ],
   "source": [
    "executeQuery(\"Does metformin interfere thyroxine absorption?\")\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 62,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Result #1\n",
      "\n",
      "------\n",
      "['Isolated ectopia lentis (EL) is caused by mutation in genes:\\n1) ADAMTSL4 and \\n2) Fibrillin-1 (FBN1).']\n",
      "\n",
      "\n",
      "\n",
      "A founder mutation in ADAMTSL4 causes early-onset bilateral ectopia lentis among Jews of Bukharian origin.\n",
      "\n",
      "\n",
      "The term isolated ectopia lentis (EL; subluxation or dislocation of the human crystalline lens) is applied to patients with EL, without skeletal features and in the absence of aortic root dilatation. To date, the only gene shown to cause autosomal-recessive isolated EL is ADAMTSL4. Here we report a novel founder mutation in ADAMTSL4 gene in children of Bukharian Jewish origin presenting with early-onset bilateral EL. \n",
      "\n",
      "\n",
      "Early onset ectopia lentis due to a FBN1 mutation with non-penetrance.\n",
      "\n",
      "\n",
      "Isolated ectopia lentis is usually autosomal dominant and commonly due to the mutations of FBN1 gene. \n",
      "\n",
      "\n",
      "In conclusion, we report on a case of early-onset autosomal dominant isolated ectopia lentis caused by FBN1 mutation that has previously been reported only in Marfan syndrome.\n",
      "\n",
      "\n",
      "ADAMTSL4-associated isolated ectopia lentis: Further patients, novel mutations and a detailed phenotype description.\n",
      "\n",
      "\n",
      "ADAMTSL4 mutations seem to be the most common cause of isolated ectoplia lentis (EL) and thus are important concerning the differential diagnosis of connective tissue syndromes with EL as main feature. \n",
      "\n",
      "\n",
      "A genotype-phenotype comparison of ADAMTSL4 and FBN1 in isolated ectopia lentis.\n",
      "\n",
      "\n",
      "To describe the genotype-phenotype relationship of a cohort of consecutive patients with isolated ectopia lentis (EL) secondary to ADAMTSL4 and FBN1 mutations.\n",
      "\n",
      "\n",
      "ADAMTSL4 is the most important known causative gene in isolated EL. Mutations in ADAMTSL4 appear to cause earlier manifestation of EL and are associated with increased axial length as compared to FBN1. We suggest that ADAMTSL4 be screened in all patients with isolated EL and that physicians be vigilant for the more severe ocular phenotype associated with mutations in this gene.\n",
      "\n",
      "\n",
      "A novel FBN1 mutation in a Chinese family with isolated ectopia lentis.\n",
      "\n",
      "\n",
      "To identify the genetic defect in an autosomal dominant isolated ectopia lentis (EL) family.\n",
      "\n",
      "\n",
      "A novel mutation of FBN1 results in an arginine to cysteine residue (p.R974C) substitution, which is responsible for the patients with isolated EL in this Chinese family.\n",
      "\n",
      "\n",
      "A homozygous mutation in ADAMTSL4 causes autosomal-recessive isolated ectopia lentis.\n",
      "\n",
      "\n",
      "Isolated ectopia lentis is usually autosomal dominant and commonly due to the mutations of FBN1 gene.\n",
      "\n",
      "\n",
      "To examine the fibrillin-1 (FBN1) gene for mutations in members of a Chinese family with isolated ectopia lentis.\n",
      "\n",
      "\n",
      "Ophthalmic findings in a family with early-onset isolated ectopia lentis and the p.Arg62Cys mutation of the fibrillin-1 gene (FBN1).\n",
      "\n",
      "\n",
      "Craniosynostosis with ectopia lentis and a homozygous 20-base deletion in ADAMTSL4.\n",
      "\n",
      "\n",
      "ADAMTSL4-associated isolated ectopia lentis: Further patients, novel mutations and a detailed phenotype description\n",
      "\n",
      "\n",
      "Our study presented detailed clinical manifestations, including some novel ophthalmic findings, such as pupillary abnormality, different types of glaucoma, and progressive hyperopia.Ophthalmic findings and the p.Arg62Cys mutation of FBN1 gene were reported in a family with early-onset isolated ectopia lentis\n",
      "\n",
      "\n",
      "To examine the fibrillin-1 (FBN1) gene for mutations in members of a Chinese family with isolated ectopia lentis.Clinically relevant laboratory investigation.Family members underwent clinical examinations\n",
      "\n",
      "\n",
      "Ophthalmic findings in a family with early-onset isolated ectopia lentis and the p.Arg62Cys mutation of the fibrillin-1 gene (FBN1)\n",
      "\n",
      "\n",
      "The 65 exons of FBN1 were amplified by polymerase chain reaction and screened for mutations by a combination of denaturing high-performance liquid chromatography analysis and direct DNA sequencing.A mutation, c.184C-->T in exon 2 of FBN1, which results in substitution of arginine by cysteine at position 62 of the fibrillin-1 protein (p.R62C) in all affected family members but in none of the unaffected individuals.A recurrent mutation of FBN1 gene resulted in an arginine-to-cysteine residue (p.R62C), is responsible for the patients with isolated ectopia lentis in a Chinese family\n",
      "\n",
      "\n",
      "Isolated ectopia lentis is usually autosomal dominant and commonly due to the mutations of FBN1 gene\n",
      "\n",
      "\n",
      "CONCLUSIONS: The results emphasize the association of ADAMTSL4 null mutations with isolated ectopia lentis and the presence of a founder mutation in the European population. \n",
      "\n",
      "\n",
      "Herein we show a consanguineous family that carries a novel homozygous splice mutation IVS4-1G>A/IVS4-1G>A in ADAMTSL4 responsible for isolated autosomal recessive EL, thus confirming the involvement of this gene in this condition and underlining the major role of ADAMTS proteases in zonular fibers homeostasis.\n",
      "\n",
      "\n",
      "Two novel FBN1 mutations associated with ectopia lentis and marfanoid habitus in two Chinese families.\n",
      "\n",
      "\n",
      "Novel FBN1 mutations associated with predominant ectopia lentis and marfanoid habitus in Chinese patients.\n",
      "\n",
      "\n",
      "FBN1 mutations are associated with multiple clinical phenotypes, including Marfan syndrome (MFS), MASS phenotype, and familial ectopia lentis, but rarely with isolated aortic aneurysm and dissection.\n",
      "\n",
      "\n",
      "This was correlated with Sanger sequencing of ADAMTSL4 and FBN1 genes.Seventeen patients were examined, including one with ectopia lentis et pupillae.\n",
      "\n",
      "\n",
      "Recently, ADAMTSL4 has been shown to be responsible for isolated autosomal recessive ectopia lentis in an inbred family.\n",
      "\n",
      "\n",
      "Confirmation of ADAMTSL4 mutations for autosomal recessive isolated bilateral ectopia lentis.\n",
      "\n",
      "\n",
      "A recurrent FBN1 mutation in an autosomal dominant ectopia lentis family of Indian origin.\n",
      "\n",
      "\n",
      "We suggest that ADAMTSL4 be screened in all patients with isolated EL and that physicians be vigilant for the more severe ocular phenotype associated with mutations in this gene..\n",
      "\n",
      "\n",
      "A homozygous mutation in ADAMTSL4 causes autosomal-recessive isolated ectopia lentis.\n",
      "\n",
      "\n",
      "Recurrent FBN1 mutation (R62C) in a Chinese family with isolated ectopia lentis.\n",
      "\n",
      "\n",
      "A genotype-phenotype comparison of ADAMTSL4 and FBN1 in isolated ectopia lentis.\n",
      "\n",
      "\n",
      "Ophthalmic findings and the p.Arg62Cys mutation of FBN1 gene were reported in a family with early-onset isolated ectopia lentis..\n",
      "\n",
      "\n",
      "Ophthalmic findings in a family with early-onset isolated ectopia lentis and the p.Arg62Cys mutation of the fibrillin-1 gene (FBN1).\n",
      "\n",
      "\n",
      "A recurrent mutation of FBN1 gene resulted in an arginine-to-cysteine residue (p.R62C), is responsible for the patients with isolated ectopia lentis in a Chinese family..\n",
      "\n",
      "\n",
      "====================================================================================================\n",
      "\n",
      "Result #2\n",
      "\n",
      "------\n",
      "['Fluorescent and luminescent reporter genes have become popular tools for the real-time monitoring of gene expression in living cells:\\ngreen fluorescent protein\\nTimer\\nred fluorescent protein\\nyellow fluorescent protein\\nbeta-phycoerythrin\\ncoral fluorescent reporter protein\\nenhanced green fluorescent reporter protein\\nmCherry']\n",
      "\n",
      "\n",
      "\n",
      "Fluorescent Timer, or DsRed1-E5, is a mutant of the red fluorescent protein, dsRed, in which fluorescence shifts over time from green to red as the protein matures.\n",
      "\n",
      "\n",
      "green fluorescent protein (Gfp)\n",
      "\n",
      "\n",
      "Characterization of flavin-based fluorescent proteins: an emerging class of fluorescent reporters.\n",
      "\n",
      "\n",
      "Flavin-based fluorescent proteins (FbFPs)\n",
      "\n",
      "\n",
      "mCherry was employed as a reporter protein \n",
      "\n",
      "\n",
      "yellow fluorescent protein\n",
      "\n",
      "\n",
      "red cytoplasmic fluorescent reporter protein. \n",
      "\n",
      "\n",
      "red fluorescent reporter protein\n",
      "\n",
      "\n",
      " enhanced green fluorescent reporter protein\n",
      "\n",
      "\n",
      "Fluorescent and luminescent reporter genes have become popular tools for the real-time monitoring of gene expression in living cells. \n",
      "\n",
      "\n",
      " fluorescent reporter protein [green fluorescent protein (GFP) and DsRed2]\n",
      "\n",
      "\n",
      "coral fluorescent reporter protein \n",
      "\n",
      "\n",
      "driving a cassette for the enhanced green \"live\" fluorescent reporter protein (eGFP) \n",
      "\n",
      "\n",
      "GFP as a fluorescent reporter protein\n",
      "\n",
      "\n",
      " protein\n",
      "\n",
      "\n",
      "red fluorescent reporter protein (RFP) gene. \n",
      "\n",
      "\n",
      "The red fluorescent protein eqFP611\n",
      "\n",
      "\n",
      "the green fluorescent reporter protein gene \n",
      "\n",
      "\n",
      " red fluorescent reporter protein\n",
      "\n",
      "\n",
      "beta-phycoerythrin, a fluorescent reporter protein derived from algae\n",
      "\n",
      "\n",
      " live-cell fluorescent reporter protein, Timer\n",
      "\n",
      "\n",
      "red fluorescent reporter protein\n",
      "\n",
      "\n",
      "====================================================================================================\n",
      "\n",
      "Result #3\n",
      "\n",
      "------\n",
      "[\"A mutation in histidine decarboxylase (Hdc) gene as well as mutations in the SLITRK1 (Slit and Trk-like 1) gene have been implicated as rare genetic causes of Tourette's syndrome.\"]\n",
      "\n",
      "\n",
      "\n",
      " A rare genetic form of Tourette syndrome due to L-histidine-decarboxylase mutation, with similar features in human and rodent, has inspired new research on functional anatomy of Tourette syndrome.\n",
      "\n",
      "\n",
      "A mutation in histidine decarboxylase (Hdc), the key enzyme for the biosynthesis of histamine (HA), has been implicated as a rare genetic cause.\n",
      "\n",
      "\n",
      "Histidine decarboxylase deficiency causes tourette syndrome: parallel findings in humans and mice.\n",
      "\n",
      "\n",
      "Tourette syndrome (TS), the most severe end of the continuum of tic disorders, is substantially genetic, but causative mutations have been elusive.\n",
      "\n",
      "\n",
      "The SLITRK1 gene encodes a developmentally regulated stimulator of neurite outgrowth and previous studies have implicated rare variants in this gene in disorders in the OC spectrum, specifically Tourette syndrome (TS)\n",
      "\n",
      "\n",
      "The recent implication of the histamine decarboxylase (HDC) gene, the key enzyme in histamine production, raises the intriguing hypothesis of a possible role of histaminergic dysfunction leading to TS onset.\n",
      "\n",
      "\n",
      "we describe an analysis of linkage in a two-generation pedigree leading to the identification of a rare functional mutation in the HDC gene encoding L-histidine decarboxylase, the rate-limiting enzyme in histamine biosynthesis.\n",
      "\n",
      "\n",
      "Mutations in the gene SLITRK1 (Slit and Trk-like 1) have been reported in patients with Tourette's disorder (TD)\n",
      "\n",
      "\n",
      "Recently, two variants, including a single-base deletion resulting in a truncated protein and a 3'-untranslated-region variant altering a binding site for micro-RNA in the Slit and Trk-like 1 gene, were found to be a genetic cause of Tourette syndrome.\n",
      "\n",
      "\n",
      "Collectively, these findings support the association of rare SLITRK1 sequence variants with TS.\n",
      "\n",
      "\n",
      "====================================================================================================\n",
      "\n",
      "Result #4\n",
      "\n",
      "------\n",
      "['LINE-1 (L1), Alu, SVA']\n",
      "\n",
      "\n",
      "\n",
      "Half of the human genome is composed of repeated DNA, and some types are mobile within our genome (transposons and retrotransposons). Despite their abundance, only a small fraction of them are currently active in our genome (Long Interspersed Element-1 (LINE-1), Alu, and SVA elements).\n",
      "\n",
      "\n",
      "Since certain types of retrotransposons, particularly members of the Alu, L1, and SVA families, are still active, their recent and ongoing propagation generates a unique and important class of human genomic diversity/polymorphism\n",
      "\n",
      "\n",
      "Alu elements are the most successful SINEs (Short INterspersed Elements) in primate genomes and have reached more than 1,000,000 copies in the human genome\n",
      "\n",
      "\n",
      "1 and Alu represent the most prolific human LINE and SINE families, respectively.\n",
      "\n",
      "\n",
      "Only a few Alu elements are able to retropose, and the factors determining their retroposition capacity are poorly understood.\n",
      "\n",
      "\n",
      "====================================================================================================\n",
      "\n",
      "Result #5\n",
      "\n",
      "------\n",
      "['SPAG5 was implicated in prostate cancer, lung cancer and cervical cancer.']\n",
      "\n",
      "\n",
      "\n",
      "BACKGROUND: We conducted multiple microarray datasets analyses from clinical and xenograft tumor tissues to search for disease progression-driving oncogenes in prostate cancer (PCa). Sperm-associated antigen 5 (SPAG5) attracted our attention. SPAG5 was recently identified as an oncogene participating in lung cancer and cervical cancer progression.\n",
      "\n",
      "\n",
      "CONCLUSION: Our results collectively showed a progression-driving role of SPAG5 in PCa which can be regulated by miR-539, suggesting that miR-539/SPAG5 can serve as a potential therapeutic target for PCa.\n",
      "\n",
      "\n",
      "RESULTS: We identified that SPAG5 expression was gradually increased in PCa progression and its level was significantly associated with lymph node metastasis, clinical stage, Gleason score, and biochemical recurrence. Our results indicated that SPAG5 knockdown can drastically inhibit PCa cell proliferation, migration, and invasion in vitro and supress tumor growth and metastasis in vivo. \n",
      "\n",
      "\n",
      "miR-539 inhibits prostate cancer progression by directly targeting SPAG5.\n",
      "\n",
      "\n",
      "SPAG5 as a prognostic biomarker and chemotherapy sensitivity predictor in breast cancer: a retrospective, integrated genomic, transcriptomic, and protein analysis.\n",
      "\n",
      "\n",
      "Copy number aberration leading to SPAG5 gain or amplification and high SPAG5 transcript and SPAG5 protein concentrations were associated with shorter overall breast cancer-specific survival (METABRIC cohort [copy number aberration]: hazard ratio [HR]1·50, 95% CI 1·18-1·92, p=0·00010; METABRIC cohort [transcript]: 1·68, 1·40-2·01, p<0·0001; and Nottingham-HES-breast cancer cohort [protein]: 1·68, 1·32-2·12, p<0·0001). In multivariable analysis, high SPAG5 transcript and SPAG5 protein expression were associated with reduced breast cancer-specific survival at 10 years compared with lower concentrations (Uppsala: HR 1·62, 95% CI 1·03-2·53, p=0·036; METABRIC: 1·27, 1·02-1·58, p=0·034; untreated lymph node-negative cohort: 2·34, 1·24-4·42, p=0·0090; and Nottingham-HES: 1·73, 1·23-2·46, p=0·0020). In patients with oestrogen receptor-negative breast cancer with high SPAG5 protein expression, anthracycline-based adjuvant chemotherapy increased breast cancer-specific survival overall compared with that for patients who did not receive chemotherapy (Nottingham-oestrogen receptor-negative-ACT cohort: HR 0·37, 95% CI 0·20-0·60, p=0·0010). Multivariable analysis showed high SPAG5 transcript concentrations to be independently associated with longer distant relapse-free survival after receiving taxane plus anthracycline neoadjuvant chemotherapy (MD Anderson-NeoACT: HR 0·68, 95% CI 0·48-0·97, p=0·031). \n",
      "\n",
      "\n",
      "INTERPRETATION: SPAG5 is a novel amplified gene on Ch17q11.2 in breast cancer. The transcript and protein products of SPAG5 are independent prognostic and predictive biomarkers that might have clinical utility as biomarkers for combination cytotoxic chemotherapy sensitivity, especially in oestrogen receptor-negative breast cancer.\n",
      "\n",
      "\n",
      "SPAG5 upregulation predicts poor prognosis in cervical cancer patients and alters sensitivity to taxol treatment via the mTOR signaling pathway.\n",
      "\n",
      "\n",
      "Previously, we found that sperm-associated antigen 5 (SPAG5) was upregulated in pelvic lymph node metastasis-positive cervical cancer.\n",
      "\n",
      "\n",
      "In general, SPAG5 upregulation relates to poor prognosis in cervical cancer patients, and SPAG5 is a regulator of mTOR activity during taxol treatment in cervical cancer.\n",
      "\n",
      "\n",
      "RESULTS: We identified novel genes whose expression was upregulated in NSCLC, including SPAG5, POLH, KIF23, and RAD54L, which are associated with mitotic spindle formation, DNA repair, chromosome segregation, and dsDNA break repair, respectively.\n",
      "\n",
      "\n",
      "Another module identified among male breast cancers, regulated by SPAG5, was closely associated with proliferation.\n",
      "\n",
      "\n",
      "In our study, SPAG5 expression in cervical cancer patients was detected using quantitative real-time polymerase chain reaction, western blotting, and immunohistochemistry; cervical cancer cell function with downregulated SPAG5 in vitro was explored using tetrazolium assay, flow cytometry, and colony formation and Transwell assays. \n",
      "\n",
      "\n",
      "In patients with oestrogen receptor-negative breast cancer with high SPAG5 protein expression, anthracycline-based adjuvant chemotherapy increased breast cancer-specific survival overall compared with that for patients who did not receive chemotherapy (Nottingham-oestrogen receptor-negative-ACT cohort: HR 0·37, 95% CI 0·20-0·60, p=0·0010).\n",
      "\n",
      "\n",
      "We investigated the clinicopathological relevance of SPAG5 gene copy number aberrations, mRNA transcript expression, and protein expression and analysed the associations of SPAG5 copy number aberrations, transcript expression, and protein expression with breast cancer-specific survival, disease-free survival, distant relapse-free survival, pathological complete response, and residual cancer burden in the Nottingham discovery cohort, Uppsala cohort, METABRIC cohort, a pooled untreated lymph node-negative cohort (n=684), a multicentre combined cohort (n=5439), the Nottingham historical early stage breast cancer cohort (Nottingham-HES; n=1650), N\n",
      "\n",
      "\n",
      "In multivariable analysis, high SPAG5 transcript and SPAG5 protein expression were associated with reduced breast cancer-specific survival at 10 years compared with lower concentrations (Uppsala: HR 1·62, 95% CI 1·03-2·53, p=0·036; METABRIC: 1·27, 1·02-1·58, p=0·034; untreated lymph node-negative cohort: 2·34, 1·24-4·42, p=0·0090; and Nottingham-HES: 1·73, 1·23-2·46, p=0·0020).\n",
      "\n",
      "\n",
      "THY1 may thus be a promising novel prognostic marker for male breast cancer. Another module identified among male breast cancers, regulated by SPAG5, was closely associated with proliferation.\n",
      "\n",
      "\n",
      "Furthermore, men with THY1 positive breast cancers had significantly inferior survival. THY1 may thus be a promising novel prognostic marker for male breast cancer. Another module identified among male breast cancers, regulated by SPAG5, was closely associated with proliferation.\n",
      "\n",
      "\n",
      "SPAG5 was recently identified as an oncogene participating in lung cancer and cervical cancer progression.\n",
      "\n",
      "\n",
      "SPAG5 is a novel amplified gene on Ch17q11.2 in breast cancer.\n",
      "\n",
      "\n",
      "Copy number aberration leading to SPAG5 gain or amplification and high SPAG5 transcript and SPAG5 protein concentrations were associated with shorter overall breast cancer-specific survival (METABRIC cohort [copy number aberration]: hazard ratio [HR] 1·50, 95% CI 1·18-1·92, p=0·00010; METABRIC cohort [transcript]: 1·68, 1·40-2·01, p<0·0001; and Nottingham-HES-breast cancer cohort [protein]: 1·68, 1·32-2·12, p<0·0001).\n",
      "\n",
      "\n",
      "====================================================================================================\n",
      "\n",
      "Result #6\n",
      "\n",
      "------\n",
      "['Both protein-coding and microRNA genes related to reproduction exhibit evidence of rapid sequence evolution in the marmoset genome.']\n",
      "\n",
      "\n",
      "\n",
      "We observed positive selection in growth hormone/insulin-like growth factor genes (growth pathways), respiratory complex I genes (metabolic pathways), and genes encoding immunobiological factors and proteases (reproductive and immunity pathways). In addition, both protein-coding and microRNA genes related to reproduction exhibited evidence of rapid sequence evolution.\n",
      "\n",
      "\n",
      "In addition, both protein-coding and microRNA genes related to reproduction exhibited evidence of rapid sequence evolution.\n",
      "\n",
      "\n",
      "In addition, both protein-coding and microRNA genes related to reproduction exhibited evidence of rapid sequence evolution. \n",
      "\n",
      "\n",
      "====================================================================================================\n",
      "\n",
      "Result #7\n",
      "\n",
      "------\n",
      "['TSH receptors are expressed also in extrathyroid tissues. TSH receptors seem to be functional. Extrathyroid tissues include fibrobasts of the orbit and adipose tissue\\nThe principal tissues with TSH receptors are:\\nadippose tissue\\n orbital fibrotic tissue']\n",
      "\n",
      "\n",
      "\n",
      "GD orbital fibroblasts, which comprise a mixture of CD34(+) and CD34(-) cells, express much lower levels of Tg and TSHR\n",
      "\n",
      "\n",
      "Previously, we found that CD34(+) progenitor cells, known as fibrocytes, express functional TSHR, infiltrate the orbit, and comprise a large subset of orbital fibroblasts in TAO. \n",
      "\n",
      "\n",
      "TSH induced lipolysis in adipose tissues.\n",
      "\n",
      "\n",
      "TSH worked as a lipolytic factor in white adipose tissues,\n",
      "\n",
      "\n",
      "These fibrocytes infiltrate orbital connective tissues in thyroid-associated ophthalmopathy and express functional TSH receptor (TSHR). \n",
      "\n",
      "\n",
      "TSHR levels are higher than those in orbital fibroblasts.\n",
      "\n",
      "\n",
      "====================================================================================================\n",
      "\n",
      "Result #8\n",
      "\n",
      "------\n",
      "['The methods for assessing the robustness/reliability of the topology of the inferred phylogenetic trees are: the widely used bootstrap method and the jackknife method.', 'In contrast to other similar software, the program FreeTree (available at http://www.natur.cuni.cz/~flegr/programs/freetree or http://ijs.sgmjournals.org/content/vol51/issue3/) can also assess the robustness of the tree topology by bootstrap, jackknife or operational taxonomic unit-jackknife analysis. (PMID: 11411692)']\n",
      "\n",
      "\n",
      "\n",
      "For robustness evaluation, we apply bootstrap and jackknife techniques to input data.\n",
      "\n",
      "\n",
      "An alignment confidence score capturing robustness to guide tree uncertainty.\n",
      "\n",
      "\n",
      "We build on the widely used bootstrap method for perturbing the phylogenetic tree.\n",
      "\n",
      "\n",
      "SuperTRI: A new approach based on branch support analyses of multiple independent data sets for assessing reliability of phylogenetic inferences.\n",
      "\n",
      "\n",
      "To overcome these limitations, we propose a new approach, called SuperTRI, which is based on the branch support analyses of the independent data sets, and where the reliability of the nodes is assessed using three measures: the supertree Bootstrap percentage and two other values calculated from the separate analyses: the mean branch support (mean Bootstrap percentage or mean posterior probability) and the reproducibility index.\n",
      "\n",
      "\n",
      "In contrast to other similar software, the program FreeTree (available at http://www.natur.cuni.cz/~flegr/programs/freetree or http://ijs.sgmjournals.org/content/vol51/issue3/) can also assess the robustness of the tree topology by bootstrap, jackknife or operational taxonomic unit-jackknife analysis.\n",
      "\n",
      "\n",
      "Unlike any previous tool for inferring phylogenies from rearrangement data, TIBA uses novel methods of robustness estimation to provide support values for the edges in the inferred tree.\n",
      "\n",
      "\n",
      "Fast and accurate phylogenetic reconstruction from high-resolution whole-genome data and a novel robustness estimator.\n",
      "\n",
      "\n",
      "Moreover, whereas phylogenetic analyses from sequence data are deemed incomplete unless bootstrapping scores (a measure of confidence) are given for each tree edge, no equivalent to bootstrapping exists for rearrangement-based phylogenetic analysis.\n",
      "\n",
      "\n",
      "We also describe a novel approach to estimate the robustness of results-an equivalent to the bootstrapping analysis used in sequence-based phylogenetic reconstruction.\n",
      "\n",
      "\n",
      "The standard deviation also suggests the reliability level of the branch order.\n",
      "\n",
      "\n",
      "Assessment of the reliability of a given phylogenetic hypothesis is an important step in phylogenetic analysis.\n",
      "\n",
      "\n",
      "The reliability of the phylogenetic trees was probed with the bootstrapping of 100 replicates of the original sequence alignments.\n",
      "\n",
      "\n",
      "====================================================================================================\n",
      "\n",
      "Result #9\n",
      "\n",
      "------\n",
      "['Aicardi-Goutières syndrome (AGS) is an inflammatory disorder caused by mutations in any of six genes (TREX1, RNASEH2A, RNASEH2B, RNASEH2C, SAMHD1, and ADAR).']\n",
      "\n",
      "\n",
      "\n",
      "Aicardi-Goutières syndrome (AGS) is an inflammatory disorder caused by mutations in any of six genes (TREX1, RNASEH2A, RNASEH2B, RNASEH2C, SAMHD1, and ADAR). \n",
      "\n",
      "\n",
      "Aicardi-Goutières syndrome (AGS) is an inflammatory disorder caused by mutations in any of six genes (TREX1, RNASEH2A, RNASEH2B, RNASEH2C, SAMHD1, and ADAR)\n",
      "\n",
      "\n",
      "====================================================================================================\n",
      "\n",
      "Result #10\n",
      "\n",
      "------\n",
      "[\"R-loops, transcriptionally-induced RNA:DNA hybrids, occurring at repeat tracts (CTG)n, (CAG)n, (CGG)n, (CCG)n and (GAA)n, are associated with diseases including myotonic dystrophy, Huntington's disease, fragile X and Friedreich's ataxia.  Physiological R-loop formation at CpG island promoters can contribute to DNA replication origin specification at these regions, the most efficient replication initiation sites in mammalian cells. R-loops may also possess beneficial effects, as their widespread formation has been detected over CpG island promoters in human genes. R-loops are particularly enriched over G-rich terminator elements.\"]\n",
      "\n",
      "\n",
      "\n",
      "R-loops, transcriptionally-induced RNA:DNA hybrids, occurring at repeat tracts (CTG)n, (CAG)n, (CGG)n, (CCG)n and (GAA)n, are associated with diseases including myotonic dystrophy, Huntington's disease, fragile X and Friedreich's ataxia\n",
      "\n",
      "\n",
      "Double-R-loop formation and processing to instability was extended to the expanded C9orf72 (GGGGCC)·(GGCCCC) repeats, known to cause amyotrophic lateral sclerosis and frontotemporal dementia, providing the first suggestion through which these repeats may become unstable. \n",
      "\n",
      "\n",
      "Here we propose that physiological R-loop formation at CpG island promoters can contribute to DNA replication origin specification at these regions, the most efficient replication initiation sites in mammalian cells.\n",
      "\n",
      "\n",
      "R-loops may also possess beneficial effects, as their widespread formation has been detected over CpG island promoters in human genes\n",
      "\n",
      "\n",
      "Furthermore, we have previously shown that R-loops are particularly enriched over G-rich terminator elements\n",
      "\n",
      "\n",
      "R-loops associated with triplet repeat expansions\n",
      "\n",
      "\n",
      "Periodic DNA patrolling underlies diverse functions of Pif1 on R-loops and G-rich DNA\n",
      "\n",
      "\n",
      "Ginno et al. (2012) describe unusual sequence features at promoter CpG islands that can lead to formation of persistent RNA-DNA hybrids (R loops), which are proposed to prevent genomic DNA methylation\n",
      "\n",
      "\n",
      "R loops stimulate genetic instability of CTG.CAG repeats\n",
      "\n",
      "\n",
      " We demonstrate, using biochemical and genetic approaches, that the formation of stable RNA.DNA hybrids enhances the instability of CTG.CAG repeat tracts.\n",
      "\n",
      "\n",
      " The R-loop, previously characterized in vitro at the leading strand replication origin (OH), is isolated as a native RNA-DNA hybrid copurifying with mtDNA\n",
      "\n",
      "\n",
      "All of these R-loops arise upon generation of a G-rich RNA strand by an RNA polymerase upon transcription of a C-rich DNA template strand.\n",
      "\n",
      "\n",
      "Finally, non-denaturing bisulfite mapping of the displaced single-stranded DNA confirmed R-loop formation at the endogenous FMR1 locus and further indicated that R-loops formed over CGG repeats may be prone to structural complexities, including hairpin formation, not commonly associated with other R-loops\n",
      "\n",
      "\n",
      "We have observed that transcription through the GC-rich FMR1 5&apos;UTR region favors R-loop formation, with the nascent (G-rich) RNA forming a stable RNA:DNA hybrid with the template DNA strand, thereby displacing the non-template DNA strand\n",
      "\n",
      "\n",
      "which is rich in AT and not prone to form R-loops,\n",
      "\n",
      "\n",
      "Finally, non-denaturing bisulfite mapping of the displaced single-stranded DNA confirmed R-loop formation at the endogenous FMR1 locus and further indicated that R-loops formed over CGG repeats may be prone to structural complexities, including hairpin formation, not commonly associated with other R-loops.\n",
      "\n",
      "\n",
      "====================================================================================================\n"
     ]
    }
   ],
   "source": [
    "executeQuery(\"Which miRNAs could be used as potential biomarkers for ovarian cancer?\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.5.2"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
